## Annotation sources

The annotated variants are based on the following underlying tools and knowledge resources:

* [VEP v90](http://www.ensembl.org/info/docs/tools/vep/index.html) - Variant Effect Predictor release 90 ([GENCODE v27](https://www.gencodegenes.org/releases/) as the gene reference dataset)
* [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP) - database of non-synonymous functional predictions (`r pcgr_data$database_versions$dbnsfp$version`)
* [TCGA](https://cancergenome.nih.gov/) - The Cancer Genome Atlas (`r pcgr_data$database_versions$tcga$version`)
* [gnomAD](http://exac.broadinstitute.org/) - germline variant frequencies exome-wide (`r pcgr_data$database_versions$gnomad$version`)
* [dbSNP](http://www.ncbi.nlm.nih.gov/SNP/) - database of short genetic variants (`r pcgr_data$database_versions$dbsnp$version`)
* [1000Genomes](ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/) - germline variant (`r pcgr_data$database_versions$onekg$version`)
* [ClinVar](http://www.ncbi.nlm.nih.gov/clinvar/) - database of clinically related variants (`r pcgr_data$database_versions$clinvar$version`)
* [DoCM](http://docm.genome.wustl.edu) - database of curated mutations (`r pcgr_data$database_versions$docm$version`)
* [CIViC](http://civic.genome.wustl.edu) - clinical interpretations of variants in cancer (`r pcgr_data$database_versions$civic$version`)
* [CBMDB](http://www.cancergenomeinterpreter.org/biomarkers) - Cancer Biomarkers database (`r pcgr_data$database_versions$cbmdb$version`)
* [IntOGen catalog of driver mutations](https://www.intogen.org/downloads) - (`r pcgr_data$database_versions$intogen_mutations$version`)
* [DisGeNET](http://www.disgenet.org) - Database of gene-disease associations (`r pcgr_data$database_versions$disgenet$version`)
* [Cancer Hotspots](http://cancerhotspots.org) - a resource for statistically significant mutations in cancer (`r pcgr_data$database_versions$cancerhotspots$version`)
* [UniProt/SwissProt KnowledgeBase](http://www.uniprot.org) - resource on protein sequence and functional information (`r pcgr_data$database_versions$uniprot$version`)
* [Pfam](http://pfam.xfam.org) - database of protein families and domains (`r pcgr_data$database_versions$pfam$version`)
* [DGIdb](http://dgidb.genome.wustl.edu) - database of targeted antineoplastic drugs (`r pcgr_data$database_versions$dgidb$version`)
* [TSGene](http://bioinfo.mc.vanderbilt.edu/TSGene/) - tumor suppressor/oncogene database (`r pcgr_data$database_versions$tsgene$version`)
* [KEGG pathways](http://www.genome.jp/kegg/pathway.html) - KEGG pathway database (`r pcgr_data$database_versions$kegg$version`)

* _PCGR software version_: `r pcgr_version`
