---
title: "Cancer Predisposition Sequencing Report - `r pcg_report$sample_name`"
date: '`r strftime(Sys.time(), format = "%B %d, %Y")`'
bibliography: bibliography.bib
output: 
  html_document:
    toc: true
    theme: default
    number_sections: false
    toc_float: true
    toc_depth: 3
    fig_width: 5
    fig_height: 4
    includes:
      after_body: disclaimer_predisposition.md
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(echo = F,warning=FALSE, dpi=72, error=F, eval=T)
```

```{r settings_data_sources, child='predispose/settings_data_sources_predispose.Rmd'}
```
## Introduction

This report is intended for interpretation of inherited DNA variants implicated with cancer susceptibility and inherited cancer syndromes. Variants in *Tier 1-3* are limited to a selected set of known cancer predisposition genes (shown below), for which the report lists __ONLY__ those variants that are

1. Previously classified without conflicting interpretations in ClinVar (clinical significance - pathogenic/likely pathogenic), or
2. Unclassified variants that are **novel** or associated with **rare** germline population frequency, i.e. 
    * Minor allele frequency (MAF) < `r pcg_report[['pcgr_config']][['maf_limits']][['maf_gnomad']]`) in the user-defined population set in the gnomAD database, and
    * Minor allele frequency (MAF) < `r pcg_report[['pcgr_config']][['maf_limits']][['maf_tgp']]`) in the user-defined population set in the 1000 Genomes Project

The following list of cancer predisposition genes are included in *Tier 1-3*:

*`r paste(sort(pcg_report[['snv_indel']][['predisposition_genes']]$symbol),collapse=", ")`*


For variants associated with cancer phenotypes through genome-wide association studies (*GWAS hits*), we do NOT make any restrictions on either

* The set of candidate genes listed, or
* The population allele frequency (MAF)

```{r summary, child='predispose/summary.Rmd'}
```
<br>

## Germline SNVs/InDels

```{r conditional_tier1, child='predispose/tier1_predispose.Rmd', eval = pcg_report[['snv_indel']][['eval']]}
```

```{r conditional_tier2, child='predispose/tier2_predispose.Rmd', eval = pcg_report[['snv_indel']][['eval']]}
```

```{r conditional_tier3, child='predispose/tier3_predispose.Rmd', eval = pcg_report[['snv_indel']][['eval']]}
```

```{r conditional_gwas, child='predispose/gwas_predispose.Rmd', eval = pcg_report[['snv_indel']][['eval']]}
```


