
### Genomic biomarkers {.tabset}

- Here, variants in the sample that overlap with reported clinical biomarkers from the [database for clinical interpretations of variants in cancer, CIViC](https://civic.genome.wustl.edu/#/home) are considered. Note that several variants in the query can overlap the same existing biomarker, given that biomarkers are reported at different resolutions (variant/gene level). Total number of clinical evidence items that coincide with query variants:
    * Predisposing: __`r NROW(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['predisposing']])`__ evidence items
    * Predictive: __`r NROW(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['predictive']])`__ evidence items
    * Prognostic: __`r NROW(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['prognostic']])`__ evidence items
    * Diagnostic: __`r NROW(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['diagnostic']])`__ evidence items

<br><br>


```{r prepare_germline_data, echo=F, results='asis'}

show_germline_filters <- list()
missing_germline_items <- list()
for(type in c('diagnostic','prognostic','predictive','predisposing')){
  show_germline_filters[[type]] <- F
  missing_germline_items[[type]] <- T
  if(NROW(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][[type]]) > 0){
    show_germline_filters[[type]] <- T
    missing_germline_items[[type]] <- F
  }
}

```


#### Predisposing

<br>

```{r germline_predisposing, echo=F, results = 'asis', eval = show_germline_filters[['predisposing']]}
library(summarywidget)
library(crosstalk)

variants_germline_predisposing_shared <- crosstalk::SharedData$new(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['predisposing']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_predisposing_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predisposing_shared, ~CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_predisposing_shared, ~EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predisposing_shared, ~SYMBOL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_predisposing_shared, ~BIOMARKER_MAPPING)
  )
)

cat('The table below lists all variant-evidence item associations:',sep='\n')
htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_germline_predisposing_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
 DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','A: FDA/NCCN/ELN guidelines','B: Clinical evidence','B1: Clinical evidence: late trials','B2: Clinical evidence: early trials','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#009E73","#56B4E9", "#56B4E9","#56B4E9","#0072B2","#E69F00", "#F0E442")))
)


```

```{r germline_predisposing_missing, echo=F, results = 'asis', eval = missing_germline_items[['predisposing']]}
cat('<i>No variant-evidence item associations found.</i>',sep='\n')
cat('\n')
```

<br><br><br>

#### Predictive

<br>

```{r germline_predictive, echo=F, results = 'asis', eval = show_germline_filters[['predictive']]}
library(summarywidget)
library(crosstalk)

variants_germline_predictive_shared <- crosstalk::SharedData$new(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['predictive']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_predictive_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predictive_shared, ~CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_predictive_shared, ~EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predictive_shared, ~SYMBOL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_predictive_shared, ~BIOMARKER_MAPPING),
    crosstalk::filter_select("THERAPEUTIC_CONTEXT", "Therapeutic context", variants_germline_predictive_shared, ~THERAPEUTIC_CONTEXT)
  )
)

cat('The table below lists all variant-evidence item associations:',sep='\n')
htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_germline_predictive_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
 DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','A: FDA/NCCN/ELN guidelines','B: Clinical evidence','B1: Clinical evidence: late trials','B2: Clinical evidence: early trials','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#009E73","#56B4E9", "#56B4E9","#56B4E9","#0072B2","#E69F00", "#F0E442")))
)


```

```{r germline_predictive_missing, echo=F, results = 'asis', eval = missing_germline_items[['predictive']]}
cat('<i>No variant-evidence item associations found.</i>',sep='\n')
cat('\n')
```

<br><br><br>

#### Prognostic

<br>

```{r germline_prognostic, echo=F, results='asis', eval = show_germline_filters[['prognostic']]}

library(summarywidget)
library(crosstalk)

variants_germline_prognostic_shared <- crosstalk::SharedData$new(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['prognostic']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_prognostic_shared, ~SYMBOL),
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_prognostic_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_prognostic_shared, ~CLINICAL_SIGNIFICANCE)
  ),
  list(
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_prognostic_shared, ~EVIDENCE_LEVEL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_prognostic_shared, ~BIOMARKER_MAPPING)
  )
)
cat('The table below lists all variant-evidence item associations:',sep='\n')
htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_germline_prognostic_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','A: FDA/NCCN/ELN guidelines','B: Clinical evidence','B1: Clinical evidence: late trials','B2: Clinical evidence: early trials','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#009E73","#56B4E9", "#56B4E9","#56B4E9","#0072B2","#E69F00", "#F0E442")))
)

```

```{r germline_prognostic_missing, echo=F, results = 'asis', eval = missing_germline_items[['prognostic']]}
cat('<i>No variant-evidence item associations found.</i>',sep='\n')
cat('\n')
```

<br><br><br>

#### Diagnostic

<br>

```{r germline_diagnostic, echo=F, results='asis', eval = show_germline_filters[['diagnostic']]}

library(summarywidget)
library(crosstalk)

variants_germline_diagnostic_shared <- crosstalk::SharedData$new(cps_report[['content']][['snv_indel']][['clinical_evidence_item']][['diagnostic']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_diagnostic_shared, ~SYMBOL),
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_diagnostic_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical association", variants_germline_diagnostic_shared, ~CLINICAL_SIGNIFICANCE)

  ),
  list(
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_diagnostic_shared, ~EVIDENCE_LEVEL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_diagnostic_shared, ~BIOMARKER_MAPPING)
  )
)
cat('The table below lists all variant-evidence item associations:',sep='\n')
htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_germline_diagnostic_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','A: FDA/NCCN/ELN guidelines','B: Clinical evidence','B1: Clinical evidence: late trials','B2: Clinical evidence: early trials','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#009E73","#56B4E9", "#56B4E9","#56B4E9","#0072B2","#E69F00", "#F0E442")))
)

```

```{r germline_diagnostic_missing, echo=F, results = 'asis', eval = missing_germline_items[['diagnostic']]}
cat('<i>No variant-evidence item associations found.</i>',sep='\n')
cat('\n')
```

<br><br><br>


