




```{r prepare_classified_variant_data, echo=F, results='asis'}


show_class_filters <- list()
missing_class_items <- list()
tot_variants <- list()
for(c in c('class1','class2','class3','class4','class5')){
  show_class_filters[[c]] <- list()
  missing_class_items[[c]] <- list()
  tot_variants[[c]] <- list()
  for(m in c('Other','ClinVar')){
    tot_variants[[c]][[m]] <- 0
    show_class_filters[[c]][[m]] <- F
    missing_class_items[[c]][[m]] <- T
    tot_variants[[c]][[m]] <- NROW(cps_report[['content']][['snv_indel']][['variant_display']][[c]] %>% dplyr::filter(SOURCE == m))
    if(tot_variants[[c]][[m]] > 0){
      show_class_filters[[c]][[m]] <- T
      missing_class_items[[c]][[m]] <- F
    }
  }
}

tag_gnomad <- cps_report[['metadata']][['config']][['popgen']][['vcftag_gnomad']]
desc_gnomad <- cps_report[['metadata']][['config']][['popgen']][['popdesc_gnomad']]
formula_gnomad <- as.formula(paste0("~",rlang::sym(tag_gnomad)))



```

### <span style="color:#9E0142">Class 5 - <b>Pathogenic</b> variants</span> {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class5']][['ClinVar']]`__ variants are registered with a <i>Pathogenic</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class5']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Pathogenic</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br>

#### ClinVar

```{r class5_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class5']][['ClinVar']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class5_clinvar <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class5']], SOURCE == 'ClinVar') %>%
  dplyr::select(-c(CPSR_CLASSIFICATION,CPSR_CLASSIFICATION_SCORE,CPSR_CLASSIFICATION_DOC,CPSR_CLASSIFICATION_CODE))
variants_class5_1 <- crosstalk::SharedData$new(variants_class5_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class5_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class5_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class5_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class5_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class5_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class5_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class5_1, formula_gnomad, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class5_1, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                      "function(settings, json) {",
                      "$(this.api().table().header()).css({'background-color': '#9E0142', 'color': '#fff'});",
                      "}")
                )
        )
)

#htmltools::br()

```

<br><br>

#### Non-ClinVar

```{r class5_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class5']][['Other']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class5_other <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class5']], SOURCE == 'Other') %>%
  dplyr::select(-c(CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN, CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE))
variants_class5_2 <- crosstalk::SharedData$new(variants_class5_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class5_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class5_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class5_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class5_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_CLASSIFICATION_SCORE", "CPSR pathogenicity score", variants_class5_2, ~CPSR_CLASSIFICATION_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class5_2, formula_gnomad, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class5_2, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                      "function(settings, json) {",
                      "$(this.api().table().header()).css({'background-color': '#9E0142', 'color': '#fff'});",
                      "}")
                )
        )
)
#htmltools::br()

```


<br><br>



### <span style="color:#D53E4F">Class 4 - <b>Likely Pathogenic</b> variants</span> {.tabset .tabset-pills .nav-justified}

<br>



*  A total of n = __`r tot_variants[['class4']][['ClinVar']]`__ variants are recorded with a <i>Likely Pathogenic</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class4']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Likely Pathogenic</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>

#### ClinVar

```{r class4_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class4']][['ClinVar']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()

variants_class4_clinvar <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class4']], SOURCE == 'ClinVar') %>%
  dplyr::select(-c(CPSR_CLASSIFICATION,CPSR_CLASSIFICATION_SCORE,CPSR_CLASSIFICATION_DOC,CPSR_CLASSIFICATION_CODE))
variants_class4_1 <- crosstalk::SharedData$new(variants_class4_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class4_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class4_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class4_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class4_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class4_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class4_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class4_1, formula_gnomad, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class4_1, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                      "function(settings, json) {",
                      "$(this.api().table().header()).css({'background-color': '#D53E4F', 'color': '#fff'});",
                      "}")
                )
        )
)

```
<br><br>

#### Non-Clinvar

```{r class4_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class4']][['Other']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class4_other <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class4']], SOURCE == 'Other') %>%
  dplyr::select(-c(CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN, CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE))
variants_class4_2 <- crosstalk::SharedData$new(variants_class4_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class4_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class4_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class4_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class4_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_CLASSIFICATION_SCORE", "CPSR pathogenicity score", variants_class4_2, ~CPSR_CLASSIFICATION_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class4_2, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class4_2, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#D53E4F', 'color': '#fff'});",
                    "}")
                )
  )
)
#htmltools::br()

```


<br><br>

### <span style="color:#000000">Class 3 - <b>Variants of Uncertain Significance (VUS)</b></span> {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class3']][['ClinVar']]`__ variants are recorded with a <i>VUS</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class3']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>VUS</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>

#### ClinVar

```{r class3_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class3']][['ClinVar']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()

variants_class3_clinvar <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class3']], SOURCE == 'ClinVar') %>%
  dplyr::select(-c(CPSR_CLASSIFICATION,CPSR_CLASSIFICATION_SCORE,CPSR_CLASSIFICATION_DOC,CPSR_CLASSIFICATION_CODE))
variants_class3_1 <- crosstalk::SharedData$new(variants_class3_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class3_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class3_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class3_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class3_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class3_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class3_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class3_1, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class3_1, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#000000', 'color': '#fff'});",
                    "}")
                )
  )
)

```
<br><br>

#### Non-Clinvar

```{r class3_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class3']][['Other']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class3_other <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class3']], SOURCE == 'Other') %>%
  dplyr::select(-c(CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN, CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE))
variants_class3_2 <- crosstalk::SharedData$new(variants_class3_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class3_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class3_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class3_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class3_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_CLASSIFICATION_SCORE", "CPSR pathogenicity score", variants_class3_2, ~CPSR_CLASSIFICATION_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class3_2, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class3_2, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#000000', 'color': '#fff'});",
                    "}")
                )
  )
)
#htmltools::br()

```

<br><br>

### <span style="color:#78C679">Class 2 - <b>Likely Benign</b> variants</span> {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class2']][['ClinVar']]`__ variants are recorded with a <i>Likely Benign</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class2']][['Other']]`__ <i><b>Other</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Likely Benign</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br>

#### ClinVar

```{r class2_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class2']][['ClinVar']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()

variants_class2_clinvar <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class2']], SOURCE == 'ClinVar') %>%
  dplyr::select(-c(CPSR_CLASSIFICATION,CPSR_CLASSIFICATION_SCORE,CPSR_CLASSIFICATION_DOC,CPSR_CLASSIFICATION_CODE))
variants_class2_1 <- crosstalk::SharedData$new(variants_class2_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class2_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class2_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class2_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class2_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class2_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class2_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class2_1, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class2_1, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#78C679', 'color': '#fff'});",
                    "}")
                )
  )
)

```
<br><br>

#### Non-Clinvar

```{r class2_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class2']][['Other']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class2_other <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class2']], SOURCE == 'Other') %>%
  dplyr::select(-c(CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN, CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE))
variants_class2_2 <- crosstalk::SharedData$new(variants_class2_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class2_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class2_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class2_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class2_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_CLASSIFICATION_SCORE", "CPSR pathogenicity score", variants_class2_2, ~CPSR_CLASSIFICATION_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class2_2, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class2_2, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#78C679', 'color': '#fff'});",
                    "}")
                )
  )
)
#htmltools::br()

```

<br><br>

### <span style="color:#077009">Class 1 - <b>Benign</b> variants</span> {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class1']][['ClinVar']]`__ variants are recorded with a <i>Benign</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class1']][['Other']]`__ <i><b>Other</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Benign</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>

#### ClinVar

```{r class1_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class1']][['ClinVar']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()

variants_class1_clinvar <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class1']], SOURCE == 'ClinVar') %>%
  dplyr::select(-c(CPSR_CLASSIFICATION,CPSR_CLASSIFICATION_SCORE,CPSR_CLASSIFICATION_DOC,CPSR_CLASSIFICATION_CODE))
variants_class1_1 <- crosstalk::SharedData$new(variants_class1_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class1_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class1_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class1_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class1_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class1_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class1_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class1_1, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class1_1, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#077009', 'color': '#fff'});",
                    "}")
                )
  )
)

```
<br><br>

#### Non-Clinvar

```{r class1_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class1']][['Other']]}
library(summarywidget)
library(crosstalk)

cat('\n')
htmltools::br()


variants_class1_other <- dplyr::filter(cps_report[['content']][['snv_indel']][['variant_display']][['class1']], SOURCE == 'Other') %>%
  dplyr::select(-c(CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN, CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE))
variants_class1_2 <- crosstalk::SharedData$new(variants_class1_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class1_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class1_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class1_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class1_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_CLASSIFICATION_SCORE", "CPSR pathogenicity score", variants_class1_2, ~CPSR_CLASSIFICATION_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_class1_2, formula_gnomad, sep="",ticks = F)
  )
)



htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_class1_2, escape=F,extensions=c("Buttons","Responsive"), width = "100%",
                options = list(
                  buttons = c('csv','excel'),
                  dom = 'Bfrtip',
                  initComplete = DT::JS(
                    "function(settings, json) {",
                    "$(this.api().table().header()).css({'background-color': '#077009', 'color': '#fff'});",
                    "}")
                )
  )
)
#htmltools::br()

```

<br><br>

