
### Tier 2 - Likely pathogenic ClinVar variants

<br>
```{r prepare_tier2_data, echo=F, results='asis'}

show_tier_filters <- F
missing_tier2 <- T
if(nrow(pcg_report[['snv_indel']][['variant_display']][['tier2']]) > 0){
  show_tier_filters <- T
  missing_tier2 <- F
}


```


```{r tier2, echo=F, results = 'asis', eval = show_tier_filters}
library(summarywidget)
library(crosstalk)

cat('A total of ',nrow(pcg_report[['snv_indel']][['variant_display']][['tier2']]),' unique, germline variant(s) are found as likely pathogenic in ClinVar for the selected cancer predisposition genes:')
cat('\n')
htmltools::br()
htmltools::br()

variants_tier2_shared <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier2']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_tier2_shared, ~CONSEQUENCE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_tier2_shared, ~SYMBOL),
    crosstalk::filter_select("PHENOTYPE", "Condition(s)", variants_tier2_shared, ~PHENOTYPE)
  ),
  list(
    crosstalk::filter_select("PROTEIN_DOMAIN", "Protein domain", variants_tier2_shared, ~PROTEIN_DOMAIN),
    crosstalk::filter_slider("NFE_AF_GNOMAD", "Minor allele frequency Europeans (gnomAD-NFE)", variants_tier2_shared, ~NFE_AF_GNOMAD,sep="", ticks = F),
    crosstalk::filter_slider("GLOBAL_AF_GNOMAD", "Minor allele frequency global (gnomAD)", variants_tier2_shared, ~GLOBAL_AF_GNOMAD, sep="",ticks = F)
  )
)

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_tier2_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip'))
)

#htmltools::br()

```

```{r tier2_missing, echo=F, results = 'asis', eval = missing_tier2}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```

<br><br>
