### Tier 3 - Unclassified variants/VUS 

<br>

The unclassified, coding variants have all been assigned a composite <i>pathogenicity score</i> (coined <b>PATHSCORE</b> in the tables below) according to the prioritization scheme outlined in [CharGer](https://github.com/ding-lab/charger/) ([Huang et al., 2018](https://www.ncbi.nlm.nih.gov/pubmed/29625052)), as well as [Maxwell et al.,2016](https://www.ncbi.nlm.nih.gov/pubmed/27153395). Specifially, a cancer-specific adoption of previously established <b>ACMG</b> criteria for variant classification has been implemented ([Richards et al., 2015](https://www.ncbi.nlm.nih.gov/pubmed/25741868), [Amendola et al., 2016](https://www.ncbi.nlm.nih.gov/pubmed/27181684)). The following ACMG evidence items indicate support for pathogenic/benign variants, and have been implemented as follows (points in parenthesis indicate how they contribute to overall PATHSCORE):  

* PVS1 (<b>8</b>) - null variant (nonsense, frameshift, canonical ±1 or 2 splice sites - indicated by [VEP's LofTee plugin](https://github.com/konradjk/loftee)) in a gene where LoF is a known mechanism of disease (dominant mode of inheritance (MoI))
* PS1 (<b>7</b>) - Same amino acid change as a previously established pathogenic variant ([ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)) regardless of nucleotide change
* PSC1 (<b>4</b>) - null variant (nonsense, frameshift, canonical ±1 or 2 splice sites - indicated by [VEP's LofTee plugin](https://github.com/konradjk/loftee)) in a gene where LoF is a known mechanism of disease (recessive MoI)
* PMC1 (<b>2</b>) - null variant (nonsense, frameshift, canonical ±1 or 2 splice sites - indicated by [VEP's LofTee plugin](https://github.com/konradjk/loftee)) in a gene where LoF is NOT the known mechanism of disease
* PM1 (<b>2</b>) - Located in a somatic mutational hotspot (cancerhotspots.org) 
* PM2 (<b>2</b>) - Absence/extremely low frequency (MAF < 0.0005 in [1000 Genomes Project](http://www.internationalgenome.org/)/[gnomAD](http://gnomad.broadinstitute.org/) global population)
* PM4 (<b>2</b>) - Protein length changes due to inframe indels or stoploss variants in known susceptibility gene (dominant MoI)
* PM5 (<b>2</b>) - Novel missense change at an amino acid residue where a different missense change determined to be pathogenic has been seen before  ([ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/))
* PP2 (<b>1</b>) - Missense variant in a gene that has a relatively low rate of benign missense variation and where missense variants are a common mechanism of disease
* PP3 (<b>1</b>) - Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc., from [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP))
* PPC1 (<b>1</b>) - Protein length changes due to inframe indels or stoploss variants in known susceptibility gene (recessive MoI)
* BP4 (<b>-1</b>) - Multiple lines (>1) of in silico evidence of none deleterious effect.
* BMC1 (<b>-2</b>) - Peptide change is at the same location of a known benign change ([ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/))
* BSC1 (<b>-6</b>) - Peptide change is known to be benign ([ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/))
* BA1 (<b>-8</b>) - High allele frequency in the general population (MAF > 0.05 in [1000 Genomes Project](http://www.internationalgenome.org/)/[gnomAD](http://gnomad.broadinstitute.org/) global population)
    * Exception for susceptibility genes HPE and SERPINA1, requiring MAF > 0.25


The composite PATHSCORE is finally assigned one of four different <b>PATHRANK</b> levels:

* <i>HIGH</i>: PATHSCORE > 8
* <i>MODERATE</i>: PATHSCORE <= 8 AND PATHSCORE > 4
* <i>LOW</i>: PATHSCORE <= 4 AND PATHSCORE >= 0
* <i>BENIGN</i>: PATHSCORE < 0

The contribution of evidence items pr. variant can be seen in the <b>PATHDOC</b> variable.


<br>

#### A: Variants registered as VUS in ClinVar {.tabset}

<br>

```{r prepare_tier3_data, echo=F, results='asis'}


show_tier3_filters <- list()
missing_tier3_items <- list()
for(type in c('cancer_phenotype','noncancer_phenotype','unclassified')){
  show_tier3_filters[[type]] <- F
  missing_tier3_items[[type]] <- T
  if(type != 'unclassified'){
    if(NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][[type]]) > 0){
      show_tier3_filters[[type]] <- T
      missing_tier3_items[[type]] <- F
    }
  }else{
    if(NROW(pcg_report[['snv_indel']][['variant_display']][['tier3B']]) > 0){
      show_tier3_filters[[type]] <- T
      missing_tier3_items[[type]] <- F
    }
  }
}

tot_variants_vus_clinvar <- NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['cancer_phenotype']]) + NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['noncancer_phenotype']])

tag_tgp <- pcg_report[['pcgr_config']][['popgen']][['vcftag_tgp']]
desc_tgp <- pcg_report[['pcgr_config']][['popgen']][['popdesc_tgp']]
formula_tgp <- as.formula(paste0("~",rlang::sym(tag_tgp)))
tag_gnomad <- pcg_report[['pcgr_config']][['popgen']][['vcftag_gnomad']]
desc_gnomad <- pcg_report[['pcgr_config']][['popgen']][['popdesc_gnomad']]
formula_gnomad <- as.formula(paste0("~",rlang::sym(tag_gnomad)))


```

A total of n = __`r tot_variants_vus_clinvar`__ variants are registered as <i>variants of uncertain significance</i> (VUS) in ClinVar.
<br><br>

##### Cancer phenotypes

```{r tier3_cancer, echo=F, results = 'asis', eval = show_tier3_filters[['cancer_phenotype']]}
library(summarywidget)
library(crosstalk)


cat('A total of <b>',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['cancer_phenotype']]), '</b> unique, germline variant(s) are registered as VUS in cancer phenotypes in ClinVar for the selected predisposition genes:')
cat('\n')
htmltools::br()
htmltools::br()

#col <- pcgr_config$custom_tags$custom_tags
#as.formula(paste0("~",rlang::sym(col)))
variants_tier3_cancer <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['cancer_phenotype']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_tier3_cancer, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_tier3_cancer, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_tier3_cancer, ~SYMBOL),
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_tier3_cancer, ~CLINVAR_PHENOTYPE)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_tier3_cancer, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_tier3_cancer, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_tier3_cancer, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_tier3_cancer, formula_tgp, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_tier3_cancer, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW','BENIGN'), c("#F46D43","#FDAE61","#FEE08B","#78C679")))
)

#htmltools::br()

```

```{r tier3_cancer_missing, echo=F, results = 'asis', eval = missing_tier3_items[['cancer_phenotype']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```
<br><br>

##### Undefined/non-cancer phenotypes

```{r tier3_noncancer, echo=F, results = 'asis', eval = show_tier3_filters[['noncancer_phenotype']]}
library(summarywidget)
library(crosstalk)


cat('A total of <b>',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['noncancer_phenotype']]), '</b> unique, germline variant(s) are registered as VUS in undefined/non-cancer phenotypes in ClinVar for the selected predisposition genes:')
cat('\n')
htmltools::br()
htmltools::br()

variants_tier3_noncancer <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['noncancer_phenotype']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_tier3_noncancer, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_tier3_noncancer, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_tier3_noncancer, ~SYMBOL),
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar henotype(s)", variants_tier3_noncancer, ~CLINVAR_PHENOTYPE)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_tier3_noncancer, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_tier3_noncancer, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_tier3_noncancer, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_tier3_noncancer, formula_tgp, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_tier3_noncancer, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW','BENIGN'), c("#F46D43","#FDAE61","#FEE08B","#78C679")))

)

#htmltools::br()

```

```{r tier3_noncancer_missing, echo=F, results = 'asis', eval = missing_tier3_items[['noncancer_phenotype']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```
<br><br>


#### B: Other unclassified protein-coding variants

<br>

```{r unclassified, echo=F, results = 'asis', eval = show_tier3_filters[['unclassified']]}
library(summarywidget)
library(crosstalk)


cat('A total of n = <b>',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3B']]),'</b> unique and protein-coding germline variant(s) are also found in the selected cancer predisposition genes. This subset is limited to variants that are **novel** (i.e. not found) or **rare**  (i.e. MAF < ',pcg_report[['pcgr_config']][['maf_limits']][['maf_gnomad']],'/',pcg_report[['pcgr_config']][['maf_limits']][['maf_tgp']],') in the ',desc_gnomad,' gnomAD population/',desc_tgp,' 1000 Genomes Project population:')
cat('\n')
    
htmltools::br()
htmltools::br()

variants_unclassified_shared <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3B']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_unclassified_shared, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_unclassified_shared, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_unclassified_shared, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_unclassified_shared, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_unclassified_shared, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_unclassified_shared, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_unclassified_shared, formula_tgp, sep="",ticks = F)
  )
)

crosstalk::bscols(
  DT::datatable(variants_unclassified_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW','BENIGN'), c("#F46D43","#FDAE61","#FEE08B","#78C679")))
)

```

```{r unclassified_missing, echo=F, results = 'asis', eval = missing_tier3_items[['unclassified']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```

<br><br>

<br>

