### Tier 3 - Unclassified variants/VUS 

<br>

#### A: Variants registered as VUS in ClinVar {.tabset}

<br>

```{r prepare_tier3_data, echo=F, results='asis'}


show_tier3_filters <- list()
missing_tier3_items <- list()
for(type in c('cancer_phenotype','noncancer_phenotype','unclassified')){
  show_tier3_filters[[type]] <- F
  missing_tier3_items[[type]] <- T
  if(type != 'unclassified'){
    if(NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][[type]]) > 0){
      show_tier3_filters[[type]] <- T
      missing_tier3_items[[type]] <- F
    }
  }else{
    if(NROW(pcg_report[['snv_indel']][['variant_display']][['tier3B']]) > 0){
      show_tier3_filters[[type]] <- T
      missing_tier3_items[[type]] <- F
    }
  }
}


tag_tgp <- pcg_report[['pcgr_config']][['popgen']][['vcftag_tgp']]
desc_tgp <- pcg_report[['pcgr_config']][['popgen']][['popdesc_tgp']]
formula_tgp <- as.formula(paste0("~",rlang::sym(tag_tgp)))
tag_gnomad <- pcg_report[['pcgr_config']][['popgen']][['vcftag_gnomad']]
desc_gnomad <- pcg_report[['pcgr_config']][['popgen']][['popdesc_gnomad']]
formula_gnomad <- as.formula(paste0("~",rlang::sym(tag_gnomad)))


```


##### Cancer phenotypes

```{r tier3_cancer, echo=F, results = 'asis', eval = show_tier3_filters[['cancer_phenotype']]}
library(summarywidget)
library(crosstalk)


cat('A total of ',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['cancer_phenotype']]), ' unique, germline variant(s) are registered as VUS in cancer phenotypes in ClinVar for the selected predisposition genes:')
cat('\n')
htmltools::br()
htmltools::br()

#col <- pcgr_config$custom_tags$custom_tags
#as.formula(paste0("~",rlang::sym(col)))
variants_tier3_cancer <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['cancer_phenotype']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_tier3_cancer, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_tier3_cancer, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_tier3_cancer, ~SYMBOL),
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_tier3_cancer, ~CLINVAR_PHENOTYPE)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_tier3_cancer, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_tier3_cancer, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_tier3_cancer, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_tier3_cancer, formula_tgp, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_tier3_cancer, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW'), c("#F46D43","#FDAE61","#FEE08B")))
)

#htmltools::br()

```

```{r tier3_cancer_missing, echo=F, results = 'asis', eval = missing_tier3_items[['cancer_phenotype']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```
<br><br>

##### Undefined/non-cancer phenotypes

```{r tier3_noncancer, echo=F, results = 'asis', eval = show_tier3_filters[['noncancer_phenotype']]}
library(summarywidget)
library(crosstalk)


cat('A total of ',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['noncancer_phenotype']]), ' unique, germline variant(s) are registered as VUS in undefined/non-cancer phenotypes in ClinVar for the selected predisposition genes:')
cat('\n')
htmltools::br()
htmltools::br()

variants_tier3_noncancer <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3A']][['noncancer_phenotype']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_tier3_noncancer, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_tier3_cancer, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_tier3_noncancer, ~SYMBOL),
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar henotype(s)", variants_tier3_noncancer, ~CLINVAR_PHENOTYPE)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_tier3_noncancer, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_tier3_noncancer, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_tier3_noncancer, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_tier3_noncancer, formula_tgp, sep="",ticks = F)
  )
)

    

htmltools::br()
htmltools::br()

crosstalk::bscols(
  DT::datatable(variants_tier3_noncancer, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW'), c("#F46D43","#FDAE61","#FEE08B")))

)

#htmltools::br()

```

```{r tier3_noncancer_missing, echo=F, results = 'asis', eval = missing_tier3_items[['noncancer_phenotype']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```
<br><br>


#### B: Other unclassified protein-coding variants

<br>

```{r unclassified, echo=F, results = 'asis', eval = show_tier3_filters[['unclassified']]}
library(summarywidget)
library(crosstalk)


cat('A total of ',NROW(pcg_report[['snv_indel']][['variant_display']][['tier3B']]),' unique and protein-coding germline variant(s) are also found in the selected cancer predisposition genes. This subset is limited to variants that are **novel** (i.e. not found) or **rare**  (i.e. MAF < ',pcg_report[['pcgr_config']][['maf_limits']][['maf_gnomad']],'/',pcg_report[['pcgr_config']][['maf_limits']][['maf_tgp']],') in the ',desc_gnomad,' gnomAD population/',desc_tgp,' 1000 Genomes Project population:')
cat('\n')
    
htmltools::br()
htmltools::br()

variants_unclassified_shared <- crosstalk::SharedData$new(pcg_report[['snv_indel']][['variant_display']][['tier3B']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_unclassified_shared, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_unclassified_shared, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_unclassified_shared, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("PATHRANK", "Pathogenicity rank", variants_unclassified_shared, ~PATHRANK),
    crosstalk::filter_checkbox("LOSS_OF_FUNCTION", "Predicted loss-of-function variant", variants_unclassified_shared, ~LOSS_OF_FUNCTION),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (",desc_gnomad,")"), variants_unclassified_shared, formula_gnomad, sep="",ticks = F),
    crosstalk::filter_slider(tag_tgp, paste0("MAF 1000 Genomes Project (",desc_tgp,")"), variants_unclassified_shared, formula_tgp, sep="",ticks = F)
  )
)

crosstalk::bscols(
  DT::datatable(variants_unclassified_shared, escape=F,extensions=c("Buttons","Responsive"), width = "100%",options=list(buttons = c('csv','excel'),dom = 'Bfrtip')) %>%
  DT::formatStyle('PATHRANK', backgroundColor = DT::styleEqual(c('HIGH','MODERATE','LOW'), c("#F46D43","#FDAE61","#FEE08B")))
)

```

```{r unclassified_missing, echo=F, results = 'asis', eval = missing_tier3_items[['unclassified']]}
cat('<i>No variants found.</i>',sep='\n')
cat('\n')
```

<br><br>

<br>

