## Somatic SNVs/InDels

### Summary statistics

* Number of SNVs: __`r nrow(report_data$sample_calls_SNVs)`__
* Number of InDels: __`r nrow(report_data$sample_calls_INDELs)`__
* Number of coding variants: __`r nrow(report_data$sample_calls_coding)`__
* Number of noncoding variants: __`r nrow(report_data$sample_calls_noncoding)`__
<br>

### Tier statistics

* Tier 1 variants: __`r nrow(report_data$variants_tier1_display)`__
* Tier 2 variants: __`r nrow(report_data$variants_tier2_display)`__
* Tier 3 variants: __`r nrow(report_data$variants_tier3_display)`__
* Tier 4 variants: __`r nrow(report_data$variants_tier4_display)`__
* Tier 5 variants: __`r nrow(report_data$variants_tier5_display)`__
<br>


### Tier 1 - Genomic biomarkers for diagnosis, prognosis, and drug sensitivity {.tabset}

- A total of __`r nrow(report_data$variants_tier1_display)`__ unique, somatic variant(s) in the tumor sample can be mapped to genomic biomarkers in the [database for clinical interpretations of variants in cancer, CIViC](https://civic.genome.wustl.edu/#/home) or [Cancer bioMarkers database](https://www.cancergenomeinterpreter.org/biomarkers), with the following number of evidence items:
    * Tier 1A: __`r nrow(report_data$clinical_evidence_items_tier1A)`__ evidence items linked to specific amino acid changes
    * Tier 1B: __`r nrow(report_data$clinical_evidence_items_tier1B)`__ evidence items linked to codon alterations
    * Tier 1C: __`r nrow(report_data$clinical_evidence_items_tier1C)`__ evidence items linked to exon alterations

<br><br>

#### Tier 1A biomarkers

<br>
```{r tier1A, echo=F, results = 'asis'}

if(nrow(report_data$clinical_evidence_items_tier1A) > 0){
  cat('The table below lists all variant-evidence item associations:',sep='\n')
  cat('\n')
  myOptions <- list(paging = F, searching=T,caching=F)
  if(nrow(report_data$clinical_evidence_items_tier1A) > 10){
    myOptions <- list(paging = T,pageLength=10, searching=T,caching=F)
  }
  report_data$clinical_evidence_items_tier1A <- dplyr::rename(report_data$clinical_evidence_items_tier1A, DOCM_DISEASE = OTHER_DISEASE_DOCM, DOCM_LITERATURE = OTHER_LITERATURE_DOCM)
  if(nrow(report_data$clinical_evidence_items_tier1A[!is.na(report_data$clinical_evidence_items_tier1A$INTOGEN_DRIVER_MUT),]) > 0){
    report_data$clinical_evidence_items_tier1A[!is.na(report_data$clinical_evidence_items_tier1A$INTOGEN_DRIVER_MUT),]$INTOGEN_DRIVER_MUT <- TRUE
  }
  DT::datatable(report_data$clinical_evidence_items_tier1A, options = myOptions, escape=F, extensions = "Responsive") %>%
  DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','B: Clinical evidence','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#56B4E9", "#0072B2","#E69F00", "#F0E442")))
}else{
  cat('<i>No variant-evidence item associations found.</i>',sep='\n')
  cat('\n')
}
```

#### Tier 1B biomarkers

<br>

```{r tier1B, echo=F, results='asis'}

if(nrow(report_data$clinical_evidence_items_tier1B) > 0){
  cat('The table below lists all variant-evidence item associations:',sep='\n')
  cat('\n')
  myOptions <- list(paging = F, searching=T,caching=F)
  if(nrow(report_data$clinical_evidence_items_tier1B) > 10){
    myOptions <- list(paging = T,pageLength=10, searching=T,caching=F)
  }
  report_data$clinical_evidence_items_tier1B <- dplyr::rename(report_data$clinical_evidence_items_tier1B, DOCM_DISEASE = OTHER_DISEASE_DOCM, DOCM_LITERATURE = OTHER_LITERATURE_DOCM)
  if(nrow(report_data$clinical_evidence_items_tier1B[!is.na(report_data$clinical_evidence_items_tier1B$INTOGEN_DRIVER_MUT),]) > 0){
    report_data$clinical_evidence_items_tier1B[!is.na(report_data$clinical_evidence_items_tier1B$INTOGEN_DRIVER_MUT),]$INTOGEN_DRIVER_MUT <- TRUE
  }
  DT::datatable(report_data$clinical_evidence_items_tier1B, options = myOptions, escape=F, extensions = "Responsive") %>%
  DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','B: Clinical evidence','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#56B4E9", "#0072B2","#E69F00", "#F0E442")))
}else{
  cat('<i>No variant-evidence item associations found.</i>',sep='\n')
  cat('\n')
}
```

#### Tier 1C biomarkers

```{r tier1C, echo=F,results='asis'}

if(nrow(report_data$clinical_evidence_items_tier1C) > 0){
  cat('The table below lists all variant-evidence item associations:',sep='\n')
  cat('\n')
  
  myOptions <- list(paging = F, searching=T,caching=F)
  if(nrow(report_data$clinical_evidence_items_tier1C) > 10){
    myOptions <- list(paging = T,pageLength=10, searching=T,caching=F)
  }
  report_data$clinical_evidence_items_tier1C <- dplyr::rename(report_data$clinical_evidence_items_tier1C, DOCM_DISEASE = OTHER_DISEASE_DOCM, DOCM_LITERATURE = OTHER_LITERATURE_DOCM)
  if(nrow(report_data$clinical_evidence_items_tier1C[!is.na(report_data$clinical_evidence_items_tier1C$INTOGEN_DRIVER_MUT),]) > 0){
    report_data$clinical_evidence_items_tier1C[!is.na(report_data$clinical_evidence_items_tier1C$INTOGEN_DRIVER_MUT),]$INTOGEN_DRIVER_MUT <- TRUE
  }

  DT::datatable(report_data$clinical_evidence_items_tier1C, options = myOptions, escape=F, extensions = "Responsive") %>%
  DT::formatStyle('EVIDENCE_LEVEL', backgroundColor = DT::styleEqual(c('A: Validated','B: Clinical evidence','C: Case study','D: Preclinical evidence','E: Indirect evidence'), c("#009E73","#56B4E9", "#0072B2","#E69F00", "#F0E442")))
}else{
  cat('<i>No variant-evidence item associations found.</i>',sep='\n')
  cat('\n')
}
```

<br><br>
