

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Annotation resources &mdash; Personal Cancer Genome Reporter 0.4.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Personal Cancer Genome Reporter 0.4.0 documentation" href="index.html"/>
        <link rel="next" title="Input &amp; output" href="output.html"/>
        <link rel="prev" title="Getting started" href="getting_started.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.4.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Annotation resources</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#basic-variant-consequence-annotation">Basic variant consequence annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#insilico-predictions-of-effect-of-coding-variants"><em>Insilico</em> predictions of effect of coding variants</a></li>
<li class="toctree-l2"><a class="reference internal" href="#variant-frequency-databases">Variant frequency databases</a></li>
<li class="toctree-l2"><a class="reference internal" href="#variant-databases-of-clinical-utility">Variant databases of clinical utility</a></li>
<li class="toctree-l2"><a class="reference internal" href="#protein-domains-functional-features">Protein domains/functional features</a></li>
<li class="toctree-l2"><a class="reference internal" href="#cancer-gene-knowledge-bases">Cancer gene knowledge bases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#notes-on-variant-annotation-datasets">Notes on variant annotation datasets</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#genome-mapping">Genome mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="#other-data-quality-concerns">Other data quality concerns</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Input &amp; output</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Personal Cancer Genome Reporter</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Annotation resources</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/annotation_resources.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="annotation-resources">
<h1>Annotation resources<a class="headerlink" href="#annotation-resources" title="Permalink to this headline">¶</a></h1>
<div class="section" id="basic-variant-consequence-annotation">
<h2>Basic variant consequence annotation<a class="headerlink" href="#basic-variant-consequence-annotation" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://www.ensembl.org/info/docs/tools/vep/index.html">VEP v85</a> -
Variant Effect Predictor release 85 (<a class="reference external" href="https://www.gencodegenes.org/releases/19.html">GENCODE
v19</a> as gene
reference database)</li>
</ul>
</div>
<div class="section" id="insilico-predictions-of-effect-of-coding-variants">
<h2><em>Insilico</em> predictions of effect of coding variants<a class="headerlink" href="#insilico-predictions-of-effect-of-coding-variants" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="https://sites.google.com/site/jpopgen/dbNSFP">dBNSFP v3.4</a> -
database of non-synonymous functional predictions (March 2017)</li>
<li><a class="reference external" href="https://www.intogen.org/downloads">IntOGen catalogs of driver
mutations/genes</a> - (May 2016)</li>
</ul>
</div>
<div class="section" id="variant-frequency-databases">
<h2>Variant frequency databases<a class="headerlink" href="#variant-frequency-databases" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://cancer.sanger.ac.uk/cosmic/">COSMIC v81</a> - catalogue of
somatic mutations in cancer (May 2017)</li>
<li><a class="reference external" href="https://dcc.icgc.org/">ICGC v23</a> - Somatic mutations discovered
in all ICGC (International Cancer Genomics Consortium) tumor cohorts
(Dec 2016)</li>
<li><a class="reference external" href="http://exac.broadinstitute.org/">ExAC r1</a> - germline variant
frequencies exome-wide (February 2017)</li>
<li><a class="reference external" href="http://exac.broadinstitute.org/">gnomAD r1</a> - germline variant
frequencies exome-wide (March 2017)</li>
<li><a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP b147</a> - database of short
genetic variants (April 2016)</li>
<li><a class="reference external" href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">1000Genomes
phase3</a>
- germline variant frequencies genome-wide (May 2013)</li>
<li><a class="reference external" href="http://cancerhotspots.org">Cancer Hotspots</a> - a resource for
statistically significant mutations in cancer (2016)</li>
</ul>
</div>
<div class="section" id="variant-databases-of-clinical-utility">
<h2>Variant databases of clinical utility<a class="headerlink" href="#variant-databases-of-clinical-utility" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a> - database of
clinically related variants (March 2017)</li>
<li><a class="reference external" href="http://docm.genome.wustl.edu">DoCM</a> - database of curated
mutations (v3.2, April 2016)</li>
<li><a class="reference external" href="http://civic.genome.wustl.edu">CIViC</a> - clinical interpretations
of variants in cancer (August 3rd 2017)</li>
<li><a class="reference external" href="http://www.cancergenomeinterpreter.org/biomarkers">CBMDB</a> -
Cancer BioMarkers database (Februay 8th 2017)</li>
<li><a class="reference external" href="http://dgidb.genome.wustl.edu">DGIdb</a> - database of interactions
betweeen antineoplastic drugs and human proteins (v2.22, February
2016)</li>
</ul>
</div>
<div class="section" id="protein-domains-functional-features">
<h2>Protein domains/functional features<a class="headerlink" href="#protein-domains-functional-features" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://www.uniprot.org">UniProt/SwissProt KnowledgeBase 2016_09</a>
- resource on protein sequence and functional information (June 2017)</li>
<li><a class="reference external" href="http://pfam.xfam.org">Pfam v31</a> - database of protein families
and domains (March 2017)</li>
</ul>
</div>
<div class="section" id="cancer-gene-knowledge-bases">
<h2>Cancer gene knowledge bases<a class="headerlink" href="#cancer-gene-knowledge-bases" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene v2.0</a> - tumor
suppressor/oncogene database (November 2015)</li>
<li><a class="reference external" href="http://cancer.sanger.ac.uk/cosmic/">Cancer Gene Cencus</a> - (v81,
May 2017)</li>
</ul>
</div>
</div>
<div class="section" id="notes-on-variant-annotation-datasets">
<h1>Notes on variant annotation datasets<a class="headerlink" href="#notes-on-variant-annotation-datasets" title="Permalink to this headline">¶</a></h1>
<div class="section" id="genome-mapping">
<h2>Genome mapping<a class="headerlink" href="#genome-mapping" title="Permalink to this headline">¶</a></h2>
<p>A requirement for all variant annotation datasets used in PCGR is that
they have been mapped unambiguously to the human genome (GRCh37). For
most datasets this is already the case (i.e. dbSNP, COSMIC, ClinVar
etc.). A significant proportion of variants in the annotation datasets
related to clinical interpretation, CIViC and CBMDB, is however not
mapped to the genome. Whenever possible, we have utilized
<a class="reference external" href="http://bioinformatics.mdanderson.org/transvarweb/">TransVar</a> to
identify the actual genomic variants (e.g. <em>g.chr7:140453136A&gt;T</em>) that
correspond to variants reported with other HGVS nomenclature (e.g.
<em>p.V600E</em>).</p>
</div>
<div class="section" id="other-data-quality-concerns">
<h2>Other data quality concerns<a class="headerlink" href="#other-data-quality-concerns" title="Permalink to this headline">¶</a></h2>
<p><strong>Clinical biomarkers</strong></p>
<p>Clinical biomarkers included in PCGR are limited to the following:</p>
<ul class="simple">
<li>Markers reported at the variant level (e.g. <strong>BRAF p.V600E</strong>)</li>
<li>Markers reported at the codon level (e.g. <strong>KRAS p.G12</strong>)</li>
<li>Markers reported at the exon level (e.g. <strong>KIT exon 11 mutation</strong>)</li>
<li>Within the <a class="reference external" href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer bioMarkers database
(CBMDB)</a>, only
markers collected from FDA/NCCN guidelines, scientific literature,
and clinical trials are included (markers collected from conference
abstracts etc. are not included)</li>
</ul>
<p><strong>COSMIC variants</strong></p>
<p>The COSMIC dataset that is part of the PCGR annotation bundle is the
subset of variants that satisfy the following criteria:</p>
<ul class="simple">
<li><strong>Mutation somatic status</strong> is either &#8216;<em>confirmed_somatic</em>&#8216; or
&#8216;<em>reported_in_another_cancer_sample_as_somatic</em>&#8216;.</li>
<li><strong>Site/histology</strong> must be known and the sample must come from a
malignant tumor (i.e. not polyps/adenomas, which are also found in
COSMIC)</li>
</ul>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="output.html" class="btn btn-neutral float-right" title="Input &amp; output" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="getting_started.html" class="btn btn-neutral" title="Getting started" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.4.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>