

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Getting started &mdash; Personal Cancer Genome Reporter 0.4.1 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Personal Cancer Genome Reporter 0.4.1 documentation" href="index.html"/>
        <link rel="next" title="Annotation resources" href="annotation_resources.html"/>
        <link rel="prev" title="About" href="about.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.4.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#prerequisites">Prerequisites</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#installation-of-docker">Installation of Docker</a></li>
<li class="toctree-l3"><a class="reference internal" href="#python">Python</a></li>
<li class="toctree-l3"><a class="reference internal" href="#download-pcgr">Download PCGR</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#run-test-generation-of-clinical-report-for-a-cancer-genome">Run test - generation of clinical report for a cancer genome</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">Annotation resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html#notes-on-variant-annotation-datasets">Notes on variant annotation datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Input &amp; output</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Personal Cancer Genome Reporter</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Getting started</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/getting_started.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="prerequisites">
<h2>Prerequisites<a class="headerlink" href="#prerequisites" title="Permalink to this headline">¶</a></h2>
<div class="section" id="installation-of-docker">
<h3>Installation of Docker<a class="headerlink" href="#installation-of-docker" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">Running PCGR requires that Docker is set up on your host. Docker has
very complete installation instructions for different platforms:</p>
<ul class="simple">
<li>installing <a class="reference external" href="https://docs.docker.com/engine/installation/linux/">Docker on
Linux</a></li>
<li>installing <a class="reference external" href="https://docs.docker.com/engine/installation/mac/">Docker on Mac
OS</a></li>
</ul>
</li>
<li><p class="first">Check that Docker is running, e.g by typing <code class="docutils literal"><span class="pre">docker</span> <span class="pre">ps</span></code> or
<code class="docutils literal"><span class="pre">docker</span> <span class="pre">images</span></code> in your terminal window</p>
</li>
<li><p class="first"><strong>IMPORTANT</strong>: The following represent the <em>minimal</em> computing
resources that must be assigned to the Docker virtual machine:</p>
<ul class="simple">
<li>Memory: 5GB</li>
<li>CPUs: 4</li>
</ul>
<p>For Docker version 1.13 on <a class="reference external" href="https://docs.docker.com/docker-for-mac/#advanced">Mac OS
X</a> there is an
option to change CPU&#8217;s and RAM from the UI and restart Docker. This
can be found through <a class="reference external" href="https://docs.docker.com/docker-for-windows/#advanced">Docker Preferences
(Advanced)</a>
in the toolbar. Similarly for <a class="reference external" href="https://docs.docker.com/docker-for-windows/#advanced">Docker for
Windows</a>.</p>
</li>
</ul>
</div>
<div class="section" id="python">
<h3>Python<a class="headerlink" href="#python" title="Permalink to this headline">¶</a></h3>
<p>An installation of Python (version 2.7.13) is required to run PCGR.
Check that Python is installed by typing <code class="docutils literal"><span class="pre">python</span> <span class="pre">--version</span></code> in your
terminal window.</p>
</div>
<div class="section" id="download-pcgr">
<h3>Download PCGR<a class="headerlink" href="#download-pcgr" title="Permalink to this headline">¶</a></h3>
<p><strong>August 10th 2017</strong>: New release (0.4.2)</p>
<ul>
<li><p class="first">Download and unpack the <a class="reference external" href="https://github.com/sigven/pcgr/releases/latest">latest software release
(0.4.2)</a></p>
</li>
<li><p class="first">Download and unpack the data bundle (approx. 17Gb) in the PCGR
directory</p>
<ul class="simple">
<li>Download <a class="reference external" href="https://drive.google.com/file/d/0B8aYD2TJ472mY1ZnRzJ2bVVWMTQ/">the latest data
bundle</a>
from Google Drive to <code class="docutils literal"><span class="pre">~/pcgr-X.X</span></code> (replace <em>X.X</em> with the
version number, e.g. <code class="docutils literal"><span class="pre">~/pcgr-0.4.2</span></code>)</li>
<li>Decompress and untar the bundle, e.g. through the following Unix
command:
<code class="docutils literal"><span class="pre">gzip</span> <span class="pre">-dc</span> <span class="pre">pcgr.databundle.GRCh37.YYYYMMDD.tgz</span> <span class="pre">|</span> <span class="pre">tar</span> <span class="pre">xvf</span> <span class="pre">-</span></code></li>
<li><strong>IMPORTANT NOTE</strong>: The data bundle contains elements (e.g.
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/assets/COSMIC_academic_license_march2015.pdf">COSMIC</a>,
<a class="reference external" href="https://www.intogen.org/downloads">IntoGen driver mutations</a>,
<a class="reference external" href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a>) which
are freely available for non-commercial use only</li>
</ul>
<p>A <em>data/</em> folder within the <em>pcgr-X.X</em> software folder should now
have been produced</p>
</li>
<li><p class="first">Pull the <a class="reference external" href="https://hub.docker.com/r/sigven/pcgr/">PCGR Docker image -
0.4.2</a> from DockerHub
(2.99Gb) :</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">docker</span> <span class="pre">pull</span> <span class="pre">sigven/pcgr:0.4.2</span></code> (PCGR annotation engine)</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="run-test-generation-of-clinical-report-for-a-cancer-genome">
<h2>Run test - generation of clinical report for a cancer genome<a class="headerlink" href="#run-test-generation-of-clinical-report-for-a-cancer-genome" title="Permalink to this headline">¶</a></h2>
<p>A tumor sample report is generated by calling the Python script
<strong>pcgr.py</strong>, which takes the following arguments and options:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">usage</span><span class="p">:</span> <span class="n">pcgr</span><span class="o">.</span><span class="n">py</span> <span class="p">[</span><span class="o">-</span><span class="n">h</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">input_vcf</span> <span class="n">INPUT_VCF</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">input_cna</span> <span class="n">INPUT_CNA</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">logR_gain</span> <span class="n">LOGR_GAIN</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">logR_homdel</span> <span class="n">LOGR_HOMDEL</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">msig_identify</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">msig_n</span> <span class="n">MSIG_N</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">msig_normalization</span> <span class="p">{</span><span class="n">default</span><span class="p">,</span><span class="n">exome</span><span class="p">,</span><span class="n">genome</span><span class="p">,</span><span class="n">exome2genome</span><span class="p">}]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">msi_predict</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">list_noncoding</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">n_vcfanno_proc</span> <span class="n">N_VCFANNO_PROC</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">n_vep_forks</span> <span class="n">N_VEP_FORKS</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">tumor_dp_tag</span> <span class="n">TUMOR_DP_TAG</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">tumor_af_tag</span> <span class="n">TUMOR_AF_TAG</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">normal_dp_tag</span> <span class="n">NORMAL_DP_TAG</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">normal_af_tag</span> <span class="n">NORMAL_AF_TAG</span><span class="p">]</span>
           <span class="p">[</span><span class="o">--</span><span class="n">call_conf_tag</span> <span class="n">CALL_CONF_TAG</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">force_overwrite</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">version</span><span class="p">]</span>
           <span class="n">pcgr_dir</span> <span class="n">output_dir</span> <span class="n">sample_id</span>

<span class="n">Personal</span> <span class="n">Cancer</span> <span class="n">Genome</span> <span class="n">Reporter</span> <span class="p">(</span><span class="n">PCGR</span><span class="p">)</span> <span class="n">workflow</span> <span class="k">for</span> <span class="n">clinical</span> <span class="n">interpretation</span> <span class="n">of</span>
<span class="n">somatic</span> <span class="n">nucleotide</span> <span class="n">variants</span> <span class="ow">and</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">aberration</span> <span class="n">segments</span>

<span class="n">positional</span> <span class="n">arguments</span><span class="p">:</span>
<span class="n">pcgr_dir</span>              <span class="n">PCGR</span> <span class="n">base</span> <span class="n">directory</span> <span class="k">with</span> <span class="n">accompanying</span> <span class="n">data</span> <span class="n">directory</span><span class="p">,</span>
                    <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="o">~/</span><span class="n">pcgr</span><span class="o">-</span><span class="mf">0.4</span><span class="o">.</span><span class="mi">2</span>
<span class="n">output_dir</span>            <span class="n">Output</span> <span class="n">directory</span>
<span class="n">sample_id</span>             <span class="n">Tumor</span> <span class="n">sample</span><span class="o">/</span><span class="n">cancer</span> <span class="n">genome</span> <span class="n">identifier</span> <span class="o">-</span> <span class="n">prefix</span> <span class="k">for</span>
                    <span class="n">output</span> <span class="n">files</span>

<span class="n">optional</span> <span class="n">arguments</span><span class="p">:</span>
<span class="o">-</span><span class="n">h</span><span class="p">,</span> <span class="o">--</span><span class="n">help</span>            <span class="n">show</span> <span class="n">this</span> <span class="n">help</span> <span class="n">message</span> <span class="ow">and</span> <span class="n">exit</span>
<span class="o">--</span><span class="n">input_vcf</span> <span class="n">INPUT_VCF</span>
                    <span class="n">VCF</span> <span class="nb">input</span> <span class="n">file</span> <span class="k">with</span> <span class="n">somatic</span> <span class="n">query</span> <span class="n">variants</span>
                    <span class="p">(</span><span class="n">SNVs</span><span class="o">/</span><span class="n">InDels</span><span class="p">)</span><span class="o">.</span> <span class="n">Note</span><span class="p">:</span> <span class="n">GRCh37</span> <span class="ow">is</span> <span class="n">currently</span> <span class="n">the</span> <span class="n">only</span>
                    <span class="n">reference</span> <span class="n">genome</span> <span class="n">build</span> <span class="n">supported</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="kc">None</span><span class="p">)</span>
<span class="o">--</span><span class="n">input_cna</span> <span class="n">INPUT_CNA</span>
                    <span class="n">Somatic</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">alteration</span> <span class="n">segments</span> <span class="p">(</span><span class="n">tab</span><span class="o">-</span><span class="n">separated</span>
                    <span class="n">values</span><span class="p">)</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="kc">None</span><span class="p">)</span>
<span class="o">--</span><span class="n">logR_gain</span> <span class="n">LOGR_GAIN</span>
                    <span class="n">Log</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span> <span class="n">ratio</span> <span class="n">treshold</span> <span class="k">for</span> <span class="n">reporting</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">gains</span>
                    <span class="ow">in</span> <span class="n">genome</span> <span class="n">report</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="mf">0.8</span><span class="p">)</span>
<span class="o">--</span><span class="n">logR_homdel</span> <span class="n">LOGR_HOMDEL</span>
                    <span class="n">Log</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span> <span class="n">ratio</span> <span class="n">treshold</span> <span class="k">for</span> <span class="n">reporting</span> <span class="n">homozygous</span>
                    <span class="n">deletions</span> <span class="ow">in</span> <span class="n">genome</span> <span class="n">report</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="o">-</span><span class="mf">0.8</span><span class="p">)</span>
<span class="o">--</span><span class="n">msig_identify</span>       <span class="n">Identify</span> <span class="n">relative</span> <span class="n">contribution</span> <span class="n">of</span> <span class="mi">30</span> <span class="n">known</span> <span class="n">mutational</span>
                    <span class="n">signatures</span> <span class="p">(</span><span class="n">COSMIC</span><span class="p">)</span> <span class="n">through</span> <span class="n">the</span> <span class="n">deconstructSigs</span>
                    <span class="n">framework</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="kc">False</span><span class="p">)</span>
<span class="o">--</span><span class="n">msig_n</span> <span class="n">MSIG_N</span>       <span class="n">deconstructSigs</span> <span class="n">option</span><span class="p">:</span> <span class="n">number</span> <span class="n">of</span> <span class="n">mutational</span>
                    <span class="n">signatures</span> <span class="p">(</span><span class="n">to</span> <span class="n">limit</span> <span class="n">the</span> <span class="n">search</span> <span class="n">to</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="mi">6</span><span class="p">)</span>
<span class="o">--</span><span class="n">msig_normalization</span> <span class="p">{</span><span class="n">default</span><span class="p">,</span><span class="n">exome</span><span class="p">,</span><span class="n">genome</span><span class="p">,</span><span class="n">exome2genome</span><span class="p">}</span>
                    <span class="n">deconstructSigs</span> <span class="n">option</span><span class="p">:</span> <span class="nb">type</span> <span class="n">of</span> <span class="n">trimer</span> <span class="n">count</span>
                    <span class="n">normalization</span> <span class="k">for</span> <span class="n">inference</span> <span class="n">of</span> <span class="n">known</span> <span class="n">mutational</span>
                    <span class="n">signatures</span><span class="p">,</span> <span class="n">see</span> <span class="n">explanation</span> <span class="n">at</span>
                    <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">raerose01</span><span class="o">/</span><span class="n">deconstructSigs</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span>
                    <span class="n">default</span><span class="p">)</span>
<span class="o">--</span><span class="n">msi_predict</span>         <span class="n">Perform</span> <span class="n">statistical</span> <span class="n">prediction</span> <span class="n">of</span> <span class="n">tumor</span> <span class="n">microsatellite</span>
                    <span class="n">instability</span> <span class="kn">from</span> <span class="nn">distribution</span> <span class="n">of</span> <span class="n">somatic</span> <span class="n">indels</span><span class="o">/</span><span class="n">snvs</span>
                    <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="kc">False</span><span class="p">)</span>
<span class="o">--</span><span class="n">list_noncoding</span>      <span class="n">List</span> <span class="n">non</span><span class="o">-</span><span class="n">coding</span> <span class="n">variants</span> <span class="ow">in</span> <span class="n">HTML</span> <span class="n">report</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span>
                    <span class="kc">False</span><span class="p">)</span>
<span class="o">--</span><span class="n">n_vcfanno_proc</span> <span class="n">N_VCFANNO_PROC</span>
                    <span class="n">Number</span> <span class="n">of</span> <span class="n">processes</span> <span class="n">used</span> <span class="n">during</span> <span class="n">vcfanno</span> <span class="n">annotation</span>
                    <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="mi">4</span><span class="p">)</span>
<span class="o">--</span><span class="n">n_vep_forks</span> <span class="n">N_VEP_FORKS</span>
                    <span class="n">Number</span> <span class="n">of</span> <span class="n">forks</span> <span class="p">(</span><span class="o">--</span><span class="n">forks</span> <span class="n">option</span> <span class="ow">in</span> <span class="n">VEP</span><span class="p">)</span> <span class="n">used</span> <span class="n">during</span>
                    <span class="n">VEP</span> <span class="n">annotation</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="mi">4</span><span class="p">)</span>
<span class="o">--</span><span class="n">tumor_dp_tag</span> <span class="n">TUMOR_DP_TAG</span>
                    <span class="n">Tag</span> <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">INFO</span> <span class="n">column</span><span class="p">)</span> <span class="n">that</span> <span class="n">denotes</span> <span class="n">total</span> <span class="n">read</span>
                    <span class="n">depth</span> <span class="n">at</span> <span class="n">variant</span> <span class="n">site</span> <span class="ow">in</span> <span class="n">tumor</span> <span class="n">sample</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="n">_na</span><span class="p">)</span>
<span class="o">--</span><span class="n">tumor_af_tag</span> <span class="n">TUMOR_AF_TAG</span>
                    <span class="n">Tag</span> <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">INFO</span> <span class="n">column</span><span class="p">)</span> <span class="n">that</span> <span class="n">denotes</span> <span class="n">fraction</span>
                    <span class="n">of</span> <span class="n">alternate</span> <span class="n">allele</span> <span class="n">reads</span> <span class="ow">in</span> <span class="n">tumor</span> <span class="n">sample</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span>
                    <span class="n">_na</span><span class="p">)</span>
<span class="o">--</span><span class="n">normal_dp_tag</span> <span class="n">NORMAL_DP_TAG</span>
                    <span class="n">Tag</span> <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">INFO</span> <span class="n">column</span><span class="p">)</span> <span class="n">that</span> <span class="n">denotes</span> <span class="n">total</span> <span class="n">read</span>
                    <span class="n">depth</span> <span class="n">at</span> <span class="n">variant</span> <span class="n">site</span> <span class="ow">in</span> <span class="n">control</span><span class="o">/</span><span class="n">normal</span> <span class="n">sample</span>
                    <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="n">_na</span><span class="p">)</span>
<span class="o">--</span><span class="n">normal_af_tag</span> <span class="n">NORMAL_AF_TAG</span>
                    <span class="n">Tag</span> <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">INFO</span> <span class="n">column</span><span class="p">)</span> <span class="n">that</span> <span class="n">denotes</span> <span class="n">fraction</span>
                    <span class="n">of</span> <span class="n">alternate</span> <span class="n">allele</span> <span class="n">reads</span> <span class="n">control</span><span class="o">/</span><span class="n">normal</span> <span class="n">sample</span>
                    <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="n">_na</span><span class="p">)</span>
<span class="o">--</span><span class="n">call_conf_tag</span> <span class="n">CALL_CONF_TAG</span>
                    <span class="n">Tag</span> <span class="ow">in</span> <span class="nb">input</span> <span class="n">VCF</span> <span class="p">(</span><span class="n">INFO</span> <span class="n">column</span><span class="p">)</span> <span class="n">that</span> <span class="n">denotes</span> <span class="n">level</span> <span class="n">of</span>
                    <span class="n">confidence</span> <span class="ow">in</span> <span class="n">variant</span> <span class="n">call</span> <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="n">_na</span><span class="p">)</span>
<span class="o">--</span><span class="n">force_overwrite</span>     <span class="n">By</span> <span class="n">default</span><span class="p">,</span> <span class="n">the</span> <span class="n">script</span> <span class="n">will</span> <span class="n">fail</span> <span class="k">with</span> <span class="n">an</span> <span class="n">error</span> <span class="k">if</span> <span class="nb">any</span>
                    <span class="n">output</span> <span class="n">file</span> <span class="n">already</span> <span class="n">exists</span><span class="o">.</span> <span class="n">You</span> <span class="n">can</span> <span class="n">force</span> <span class="n">the</span>
                    <span class="n">overwrite</span> <span class="n">of</span> <span class="n">existing</span> <span class="n">result</span> <span class="n">files</span> <span class="n">by</span> <span class="n">using</span> <span class="n">this</span> <span class="n">flag</span>
                    <span class="p">(</span><span class="n">default</span><span class="p">:</span> <span class="kc">False</span><span class="p">)</span>
<span class="o">--</span><span class="n">version</span>             <span class="n">show</span> <span class="n">program</span><span class="s1">&#39;s version number and exit</span>
</pre></div>
</div>
<p>The <em>examples</em> folder contain input files from two tumor samples
sequenced within TCGA. A report for a colorectal tumor case can be
generated by running the following command in your terminal window:</p>
<p><code class="docutils literal"><span class="pre">python</span> <span class="pre">pcgr.py</span> <span class="pre">--input_vcf</span> <span class="pre">~/pcgr-0.4.2/examples/tumor_sample.COAD.vcf.gz</span> <span class="pre">--input_cna</span> <span class="pre">~/pcgr-0.4.2/examples/tumor_sample.COAD.cna.tsv</span></code>
<code class="docutils literal"><span class="pre">--msig_identify</span> <span class="pre">--msi_predict</span> <span class="pre">--tumor_af_tag</span> <span class="pre">TVAF</span> <span class="pre">--tumor_dp_tag</span> <span class="pre">TDP</span> <span class="pre">--call_conf_tag</span> <span class="pre">TAL</span> <span class="pre">--list_noncoding</span></code>
<code class="docutils literal"><span class="pre">~/pcgr-0.4.2</span> <span class="pre">~/pcgr-0.4.2/examples</span> <span class="pre">tumor_sample.COAD</span></code></p>
<p>This command will run the Docker-based PCGR workflow and produce the
following output files in the <em>examples</em> folder:</p>
<ol class="arabic simple">
<li><strong>tumor_sample.COAD.pcgr.html</strong> - An interactive HTML report for
clinical interpretation</li>
<li><strong>tumor_sample.COAD.pcgr.vcf.gz</strong> - VCF file with rich set of
annotations for precision oncology</li>
<li><strong>tumor_sample.COAD.pcgr.maf</strong> - A basic MAF file for use as input
in downstream analyses with other tools (e.g.
<a class="reference external" href="https://github.com/KarchinLab/2020plus">2020plus</a>, MutSigCV)</li>
<li><strong>tumor_sample.COAD.pcgr.snvs_indels.tiers.tsv</strong> - Tab-separated
values file with variants organized according to tiers of functional
relevance</li>
<li><strong>tumor_sample.COAD.pcgr.mutational_signatures.tsv</strong> -
Tab-separated values file with estimated contributions by known
mutational signatures and associated underlying etiologies</li>
<li><strong>tumor_sample.COAD.pcgr.snvs_indels.biomarkers.tsv</strong> -
Tab-separated values file with clinical evidence items associated
with biomarkers for diagnosis, prognosis or drug
sensitivity/resistance</li>
<li><strong>tumor_sample.COAD.pcgr.cna_segments.tsv.gz</strong> - Tab-separated
values file with annotations of gene transcripts that overlap with
somatic copy number aberrations</li>
</ol>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="annotation_resources.html" class="btn btn-neutral float-right" title="Annotation resources" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="about.html" class="btn btn-neutral" title="About" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.4.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>