

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Input &mdash; Personal Cancer Genome Reporter 0.6.2 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Personal Cancer Genome Reporter 0.6.2 documentation" href="index.html"/>
        <link rel="next" title="Output" href="output.html"/>
        <link rel="prev" title="Annotation resources" href="annotation_resources.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.6.2
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">Annotation resources</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Input</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#vcf">VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="#copy-number-segments">Copy number segments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pcgr-configuration-file">PCGR configuration file</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="tier_systems.html">Tier models</a></li>
<li class="toctree-l1"><a class="reference internal" href="CHANGELOG.html">CHANGELOG</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Personal Cancer Genome Reporter</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Input</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/input.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="input">
<h1>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h1>
<p>The PCGR workflow accepts two types of input files:</p>
<ul class="simple">
<li>An unannotated, single-sample VCF file (&gt;= v4.2) with called somatic
variants (SNVs/InDels)</li>
<li>A copy number segment file</li>
</ul>
<p>PCGR can be run with either or both of the two input files present.</p>
<div class="section" id="vcf">
<h2>VCF<a class="headerlink" href="#vcf" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>We <strong>strongly</strong> recommend that the input VCF is compressed and
indexed using <a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a></li>
<li>If the input VCF contains multi-allelic sites, these will be subject
to <a class="reference external" href="http://genome.sph.umich.edu/wiki/Vt#Decompose">decomposition</a></li>
<li>Variants used for reporting should be designated as &#8216;PASS&#8217; in the VCF
FILTER column</li>
</ul>
<p><strong>IMPORTANT NOTE 1</strong>: Considering the VCF output for the <a class="reference external" href="https://www.biostars.org/p/19104/">numerous
somatic SNV/InDel callers</a> that
have been developed, we have a experienced a general lack of uniformity
and robustness for the representation of somatic variant genotype data
(e.g. variant allelic depths (tumor/normal), genotype quality etc.).
Variant genotype data found as INFO tags in the input VCF can be
specified as optional arguments to the PCGR workflow, which in turn can
be used for interactive exploration in the tumor report.</p>
<p><strong>IMPORTANT NOTE 2</strong>: PCGR generates a number of VCF INFO annotation
tags that is appended to the query VCF. We will therefore encourage the
users to submit query VCF files that have not been subject to
annotations by other means, but rather a VCF file that comes directly
from variant calling. If not, there are likely to be INFO tags in the
query VCF file that coincide with those produced by PCGR.</p>
</div>
<div class="section" id="copy-number-segments">
<h2>Copy number segments<a class="headerlink" href="#copy-number-segments" title="Permalink to this headline">¶</a></h2>
<p>The tab-separated values file with copy number aberrations <strong>MUST</strong>
contain the following four columns:</p>
<ul class="simple">
<li>Chromosome</li>
<li>Start</li>
<li>End</li>
<li>Segment_Mean</li>
</ul>
<p>Here, <em>Chromosome</em>, <em>Start</em>, and <em>End</em> denote the chromosomal segment,
and <strong>Segment_Mean</strong> denotes the log(2) ratio for a particular segment,
which is a common output of somatic copy number alteration callers.
Below shows the initial part of a copy number segment file that is
formatted correctly according to PCGR&#8217;s requirements:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">Chromosome</span>    <span class="n">Start</span>   <span class="n">End</span> <span class="n">Segment_Mean</span>
<span class="mi">1</span> <span class="mi">3218329</span> <span class="mi">3550598</span> <span class="mf">0.0024</span>
<span class="mi">1</span> <span class="mi">3552451</span> <span class="mi">4593614</span> <span class="mf">0.1995</span>
<span class="mi">1</span> <span class="mi">4593663</span> <span class="mi">6433129</span> <span class="o">-</span><span class="mf">1.0277</span>
</pre></div>
</div>
</div>
<div class="section" id="pcgr-configuration-file">
<h2>PCGR configuration file<a class="headerlink" href="#pcgr-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>The cancer genome sequencing report can be flexibly configured in a
TOML-formatted configuration file. The default TOML configuration file,
with descriptive comments wrt. usage are shown below:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="c1"># PCGR configuration options (TOML).</span>

<span class="p">[</span><span class="n">tier_model</span><span class="p">]</span>
<span class="c1">## tier model for prioritization of SNVs/InDels (&quot;pcgr_acmg&quot; or &quot;pcgr&quot;)</span>
<span class="n">tier_model</span> <span class="o">=</span> <span class="s2">&quot;pcgr_acmg&quot;</span>

<span class="p">[</span><span class="n">tumor_only</span><span class="p">]</span>
<span class="c1">## If input VCF contains mix of germline/somatic (variants called with no matching control, i.e. tumor-only) set vcf_tumor_only to true</span>
<span class="n">vcf_tumor_only</span> <span class="o">=</span> <span class="n">false</span>

<span class="c1">## if vcf_tumor_only = true, exclude variants (SNVs/InDels) with minor allele frequency above the following population-specific thresholds</span>
<span class="c1">## 1000 Genomes Project - WGS data</span>
<span class="n">maf_onekg_eur</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_onekg_amr</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_onekg_afr</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_onekg_sas</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_onekg_eas</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_onekg_global</span> <span class="o">=</span> <span class="mf">0.01</span>

<span class="c1">## exclude variants with minor allele frequency above the following population-specific thresholds</span>
<span class="c1">## gnomAD - WES data</span>
<span class="n">maf_gnomad_nfe</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_amr</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_afr</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_sas</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_eas</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_fin</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_oth</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="n">maf_gnomad_global</span> <span class="o">=</span> <span class="mf">0.01</span>

<span class="c1">## exclude variants found in dbSNP (only those not found as somatic in ClinVar/Docm)</span>
<span class="n">exclude_dbsnp_nonclinical</span> <span class="o">=</span> <span class="n">true</span>

<span class="c1">## in variant exclusion from dbSNP, set whether found in TCGA should be kept (at desired recurrence level)</span>
<span class="c1">## E.g. keep_known_tcga = true + tcga_recurrence = 2 keeps all TCGA variants (that intersect dbSNP) found in at least two samples</span>
<span class="n">keep_known_tcga</span> <span class="o">=</span> <span class="n">true</span>
<span class="n">tcga_recurrence</span> <span class="o">=</span> <span class="mi">2</span>

<span class="c1">## exclude all non protein-coding variants</span>
<span class="n">exclude_noncoding</span> <span class="o">=</span> <span class="n">true</span>

<span class="p">[</span><span class="n">allelic_support</span><span class="p">]</span>
<span class="c1">## Specify INFO tags in input VCF that denotes depth/allelic fraction in tumor and normal sample</span>
<span class="c1">## An additional tag that denotes call confidence (call_conf_tag) can also be specified, which will</span>
<span class="c1">## be used for exploration in the global variant browser. Note that &#39;tumor_dp_tag&#39; must be of</span>
<span class="c1">## Type=Integer, and &#39;tumor_af_tag&#39; must be of Type=Float (similarly for normal sample)</span>
<span class="n">tumor_dp_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">tumor_af_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">normal_dp_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">normal_af_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">call_conf_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>

<span class="c1">## set thresholds for tumor/normal depth/allelic fraction, will be applied before report generation</span>
<span class="c1">## requires that &#39;tumor/normal_dp_tag&#39; and &#39;tumor/normal_af_tag&#39; are specified above</span>
<span class="n">tumor_dp_min</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">tumor_af_min</span> <span class="o">=</span> <span class="mf">0.0</span>
<span class="n">normal_dp_min</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">normal_af_max</span> <span class="o">=</span> <span class="mf">1.0</span>

<span class="p">[</span><span class="n">mutational_burden</span><span class="p">]</span>
<span class="c1">## Calculate mutational burden (similar to Chalmers et al., Genome Med, 2017)</span>
<span class="n">mutational_burden</span> <span class="o">=</span> <span class="n">true</span>
<span class="c1">## Size of coding target region in megabases (defaults to exome ~ 36 Mb)</span>
<span class="c1">## Note: this should ideally denote the callable target size (i.e. reflecting variable</span>
<span class="c1">## sequencing depth)</span>
<span class="n">target_size_mb</span> <span class="o">=</span> <span class="mf">36.0</span>
<span class="c1">## set upper limits to tumor mutational burden tertiles (mutations/Mb)</span>
<span class="n">tmb_low_limit</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">tmb_intermediate_limit</span> <span class="o">=</span> <span class="mi">20</span>
<span class="c1">## tmb_high = tmb &gt; tmb_intermediate_limit</span>

<span class="p">[</span><span class="n">cna</span><span class="p">]</span>
<span class="c1">## log ratio thresholds for determination of copy number gains and homozygous deletions</span>
<span class="n">logR_gain</span> <span class="o">=</span> <span class="mf">0.8</span>
<span class="n">logR_homdel</span> <span class="o">=</span> <span class="o">-</span><span class="mf">0.8</span>

<span class="c1">## percent overlap between copy number segment and gene transcripts (average) for reporting of gains/losses in tumor suppressor genes/oncogenes</span>
<span class="n">cna_overlap_pct</span> <span class="o">=</span> <span class="mi">100</span>

<span class="p">[</span><span class="n">msi</span><span class="p">]</span>
<span class="c1">## Predict microsatellite instability</span>
<span class="n">msi</span> <span class="o">=</span> <span class="n">true</span>

<span class="p">[</span><span class="n">mutational_signatures</span><span class="p">]</span>
<span class="c1">## Identify relative contribution of 30 known mutational signatures (COSMIC) through the deconstructSigs framework</span>
<span class="n">mutsignatures</span> <span class="o">=</span> <span class="n">true</span>
<span class="c1">## deconstructSigs option: number of mutational signatures to limit the search to (&#39;signatures.limit&#39; in whichSignatures)</span>
<span class="n">mutsignatures_signature_limit</span> <span class="o">=</span> <span class="mi">6</span>
<span class="c1">## deconstructSigs option: type of trimer count normalization for inference of known mutational signatures, see explanation at https://github.com/raerose01/deconstructSigs&quot;</span>
<span class="c1">## options = &#39;default&#39;, &#39;exome&#39;, &#39;genome&#39;, &#39;exome2genome&#39;</span>
<span class="c1">## NOTE: If your data (VCF) is from exome sequencing, &#39;default&#39; or &#39;exome2genome&#39; should be used. See https://github.com/raerose01/deconstructSigs/issues/2</span>
<span class="n">mutsignatures_normalization</span> <span class="o">=</span> <span class="s2">&quot;default&quot;</span>
<span class="c1">## Require a minimum number of mutations for signature estimation</span>
<span class="n">mutsignatures_mutation_limit</span> <span class="o">=</span> <span class="mi">100</span>
<span class="c1">## deconstructSigs option: discard any signature contributions with a weight less than this amount</span>
<span class="n">mutsignatures_cutoff</span> <span class="o">=</span> <span class="mf">0.06</span>

<span class="p">[</span><span class="n">tumor_type</span><span class="p">]</span>
<span class="c1">## Choose tumor type/class of input sample</span>
<span class="c1">## Due to partial overlap between some classes, user can set maximum two types</span>
<span class="n">Adrenal_Gland_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Ampullary_Carcinoma_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Biliary_Tract_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Bladder_Urinary_Tract_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Blood_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Bone_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Breast_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">CNS_Brain_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Colorectal_Cancer_NOS</span> <span class="o">=</span> <span class="n">true</span>
<span class="n">Cervical_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">DNA_Repair_Deficiency_Disorders</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Esophageal_Stomach_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Head_And_Neck_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Hereditary_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Kidney_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Leukemia_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Liver_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Lung_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Lymphoma_Hodgkin_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Lymphoma_Non_Hodgkin_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Mesothelioma</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Multiple_Myeloma</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Ovarian_Fallopian_Tube_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Pancreatic_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Penile_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Peripheral_Nervous_System_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Peritoneal_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Pleural_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Prostate_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Skin_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Soft_Tissue_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Stomach_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Testicular_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Thymic_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Thyroid_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Uterine_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>
<span class="n">Vulvar_Vaginal_Cancer_NOS</span> <span class="o">=</span> <span class="n">false</span>

<span class="p">[</span><span class="n">visual</span><span class="p">]</span>
<span class="c1">## Choose visual theme of report, any of: &quot;default&quot;, &quot;cerulean&quot;, &quot;journal&quot;, &quot;flatly&quot;, &quot;readable&quot;, &quot;spacelab&quot;, &quot;united&quot;, &quot;cosmo&quot;, &quot;lumen&quot;, &quot;paper&quot;, &quot;sandstone&quot;, &quot;simplex&quot;, or &quot;yeti&quot; (https://bootswatch.com/)</span>
<span class="n">report_theme</span> <span class="o">=</span> <span class="s2">&quot;default&quot;</span>

<span class="p">[</span><span class="n">other</span><span class="p">]</span>
<span class="c1">## Keep/skip VCF validation by https://github.com/EBIvariation/vcf-validator</span>
<span class="c1">## The vcf-validator checks that the input VCF is properly encoded. Since the</span>
<span class="c1">## vcf-validator is strict, and with error messages that is not always self-explanatory,</span>
<span class="c1">## the users can skip validation if they are confident that the most critical parts of the VCF</span>
<span class="c1">## are properly encoded</span>
<span class="n">vcf_validation</span> <span class="o">=</span> <span class="n">true</span>
<span class="c1">## list/do not list noncoding variants</span>
<span class="n">list_noncoding</span> <span class="o">=</span> <span class="n">true</span>
<span class="c1">## VEP/vcfanno processing options</span>
<span class="n">n_vcfanno_proc</span> <span class="o">=</span> <span class="mi">4</span>
<span class="n">n_vep_forks</span> <span class="o">=</span> <span class="mi">4</span>
<span class="c1">## omit intergenic variants during VEP processing</span>
<span class="n">vep_skip_intergenic</span> <span class="o">=</span> <span class="n">false</span>
</pre></div>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="output.html" class="btn btn-neutral float-right" title="Output" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="annotation_resources.html" class="btn btn-neutral" title="Annotation resources" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017/2018, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.6.2',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>