

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Input &mdash; Personal Cancer Genome Reporter 0.9.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Output" href="output.html" />
    <link rel="prev" title="PCGR annotation resources" href="annotation_resources.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.9.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">PCGR annotation resources</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Input</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#vcf">VCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="#copy-number-segments">Copy number segments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pcgr-configuration-file">PCGR configuration file</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="tier_systems.html">Tier models</a></li>
<li class="toctree-l1"><a class="reference internal" href="CHANGELOG.html">CHANGELOG</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Personal Cancer Genome Reporter</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Input</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/input.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="input">
<h1>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h1>
<p>The PCGR workflow accepts two types of input files:</p>
<ul class="simple">
<li><p>An unannotated, single-sample VCF file (&gt;= v4.2) with called somatic
variants (SNVs/InDels)</p></li>
<li><p>A copy number segment file</p></li>
</ul>
<p>PCGR can be run with either or both of the two input files present.</p>
<div class="section" id="vcf">
<h2>VCF<a class="headerlink" href="#vcf" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>We <strong>strongly</strong> recommend that the input VCF is compressed and
indexed using <a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a></p></li>
<li><p>If the input VCF contains multi-allelic sites, these will be subject
to <a class="reference external" href="http://genome.sph.umich.edu/wiki/Vt#Decompose">decomposition</a></p></li>
<li><p>Variants used for reporting should be designated as ‘PASS’ in the VCF
FILTER column</p></li>
</ul>
<p><strong>IMPORTANT NOTE 1</strong>: Considering the VCF output for the <a class="reference external" href="https://www.biostars.org/p/19104/">numerous
somatic SNV/InDel callers</a> that
have been developed, we have a experienced a general lack of uniformity
and robustness for the representation of somatic variant genotype data
(e.g. variant allelic depths (tumor/normal), genotype quality etc.).
Variant genotype data found as INFO tags in the input VCF can be
specified as optional arguments to the PCGR workflow, which in turn can
be used for interactive exploration in the tumor report.</p>
<p><strong>IMPORTANT NOTE 2</strong>: PCGR generates a number of VCF INFO annotation
tags that is appended to the query VCF. We will therefore encourage the
users to submit query VCF files that have not been subject to
annotations by other means, but rather a VCF file that comes directly
from variant calling. If not, there are likely to be INFO tags in the
query VCF file that coincide with those produced by PCGR.</p>
</div>
<div class="section" id="copy-number-segments">
<h2>Copy number segments<a class="headerlink" href="#copy-number-segments" title="Permalink to this headline">¶</a></h2>
<p>The tab-separated values file with copy number aberrations <strong>MUST</strong>
contain the following four columns:</p>
<ul class="simple">
<li><p>Chromosome</p></li>
<li><p>Start</p></li>
<li><p>End</p></li>
<li><p>Segment_Mean</p></li>
</ul>
<p>Here, <em>Chromosome</em>, <em>Start</em>, and <em>End</em> denote the chromosomal segment,
and <strong>Segment_Mean</strong> denotes the log(2) ratio for a particular segment,
which is a common output of somatic copy number alteration callers. Note
that coordinates must be <strong>one-based</strong> (i.e. chromosomes start at 1, not
0). Below shows the initial part of a copy number segment file that is
formatted correctly according to PCGR’s requirements:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Chromosome</span>    <span class="n">Start</span>   <span class="n">End</span> <span class="n">Segment_Mean</span>
<span class="mi">1</span> <span class="mi">3218329</span> <span class="mi">3550598</span> <span class="mf">0.0024</span>
<span class="mi">1</span> <span class="mi">3552451</span> <span class="mi">4593614</span> <span class="mf">0.1995</span>
<span class="mi">1</span> <span class="mi">4593663</span> <span class="mi">6433129</span> <span class="o">-</span><span class="mf">1.0277</span>
</pre></div>
</div>
</div>
<div class="section" id="pcgr-configuration-file">
<h2>PCGR configuration file<a class="headerlink" href="#pcgr-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>Some features and analysis performed in PCGR must be configured in a
TOML-formatted configuration file. The default TOML configuration file,
with descriptive comments wrt. usage are shown below:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># PCGR configuration options (TOML).</span>

<span class="p">[</span><span class="n">tumor_only</span><span class="p">]</span>
<span class="c1">## Variant filtering applied for option --tumor_only = true in pcgr.py</span>
<span class="c1">## Several filters can be configured, all as a means to minimize the proportion of germline calls in the raw set derived from tumor-only calling</span>

<span class="c1">## if vcf_tumor_only = true, exclude variants (SNVs/InDels) with minor allele frequency above the following population-specific thresholds</span>
<span class="c1">## 1000 Genomes Project - WGS data</span>
<span class="n">maf_onekg_eur</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_onekg_amr</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_onekg_afr</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_onekg_sas</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_onekg_eas</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_onekg_global</span> <span class="o">=</span> <span class="mf">0.002</span>

<span class="c1">## exclude variants with minor allele frequency above the following population-specific thresholds</span>
<span class="c1">## gnomAD - WES data</span>
<span class="n">maf_gnomad_nfe</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_amr</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_afr</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_sas</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_eas</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_fin</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_oth</span> <span class="o">=</span> <span class="mf">0.002</span>
<span class="n">maf_gnomad_global</span> <span class="o">=</span> <span class="mf">0.002</span>

<span class="c1">## Exclude variants occurring in PoN (panel of normals, if provided as VCF)</span>
<span class="n">exclude_pon</span> <span class="o">=</span> <span class="n">true</span>

<span class="c1">## Exclude likely homozygous germline variants (100% allelic fraction for alternate allele in tumor, very unlikely somatic event)</span>
<span class="n">exclude_likely_hom_germline</span> <span class="o">=</span> <span class="n">false</span>

<span class="c1">## Exclude likely heterozygous germline variants</span>
<span class="c1">## Must satisfy i) 40-60 % allelic fraction for alternate allele in tumor sample, ii) present in dbSNP + gnomAD, ii) not existing as somatic event in COSMIC/TCGA</span>
<span class="c1">## Note that the application of this filter may be suboptimal for very impure tumors or variants affected by CNAs etc (under these circumstances, the allelic fraction</span>
<span class="c1">## will be skewed (see e.g. discussion in PMID:29249243)</span>
<span class="n">exclude_likely_het_germline</span> <span class="o">=</span> <span class="n">false</span>

<span class="c1">## Exclude variants found in dbSNP (only those that are NOT found in ClinVar(somatic origin)/DoCM/TCGA/COSMIC)</span>
<span class="n">exclude_dbsnp_nonsomatic</span> <span class="o">=</span> <span class="n">false</span>

<span class="c1">## exclude all non-exonic variants</span>
<span class="n">exclude_nonexonic</span> <span class="o">=</span> <span class="n">true</span>


<span class="p">[</span><span class="n">allelic_support</span><span class="p">]</span>
<span class="c1">## Specify INFO tags in input VCF that denotes depth/allelic fraction in tumor and control sample</span>
<span class="c1">## An additional tag that denotes call confidence (call_conf_tag) can also be specified, which will</span>
<span class="c1">## be used for exploration in the global variant browser. Note that &#39;tumor_dp_tag&#39; must be of</span>
<span class="c1">## Type=Integer, and &#39;tumor_af_tag&#39; must be of Type=Float (similarly for control sample)</span>
<span class="n">tumor_dp_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">tumor_af_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">control_dp_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">control_af_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">call_conf_tag</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>

<span class="p">[</span><span class="n">visual</span><span class="p">]</span>
<span class="c1">## Choose visual theme of report, any of: &quot;default&quot;, &quot;cerulean&quot;, &quot;journal&quot;, &quot;flatly&quot;, &quot;readable&quot;, &quot;spacelab&quot;, &quot;united&quot;, &quot;cosmo&quot;, &quot;lumen&quot;, &quot;paper&quot;, &quot;sandstone&quot;, &quot;simplex&quot;, or &quot;yeti&quot; (https://bootswatch.com/)</span>
<span class="n">report_theme</span> <span class="o">=</span> <span class="s2">&quot;default&quot;</span>

<span class="p">[</span><span class="n">custom_tags</span><span class="p">]</span>
<span class="c1">## list VCF info tags that should be present in JSON and TSV output</span>
<span class="c1">## tags should be comma separated, i.e. custom_tags = &quot;MUTECT2_FILTER,STRELKA_FILTER&quot;</span>
<span class="n">custom_tags</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>

<span class="p">[</span><span class="n">other</span><span class="p">]</span>
<span class="c1">## list/do not list noncoding variants</span>
<span class="n">list_noncoding</span> <span class="o">=</span> <span class="n">true</span>
<span class="c1">## VEP/vcfanno processing options</span>
<span class="n">n_vcfanno_proc</span> <span class="o">=</span> <span class="mi">4</span>
<span class="n">n_vep_forks</span> <span class="o">=</span> <span class="mi">4</span>
<span class="c1">## Customise the order of criteria used to pick the primary transcript in VEP (see https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_pick_order)</span>
<span class="n">vep_pick_order</span> <span class="o">=</span> <span class="s2">&quot;canonical,appris,biotype,ccds,rank,tsl,length&quot;</span>
<span class="c1">## omit intergenic variants during VEP processing</span>
<span class="n">vep_skip_intergenic</span> <span class="o">=</span> <span class="n">false</span>
<span class="c1">## generate a MAF for input VCF using https://github.com/mskcc/vcf2maf</span>
<span class="n">vcf2maf</span> <span class="o">=</span> <span class="n">true</span>
</pre></div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="output.html" class="btn btn-neutral float-right" title="Output" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="annotation_resources.html" class="btn btn-neutral float-left" title="PCGR annotation resources" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2020, Sigve Nakken

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>