

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>NCGC - functional annotation of somatic DNA variation &mdash; Personal Cancer Genome Reporter 0.1 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Personal Cancer Genome Reporter 0.1 documentation" href="index.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="about.html">What is the Personal Cancer Genome Reporter (PCGR)?</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html#why-use-pcgr">Why use PCGR?</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html#docker-based-technology">Docker-based technology</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">Data sources included</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html#somatic-snvs-indels">Somatic SNVs/InDels</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html#somatic-copy-number-abberations">Somatic copy number abberations</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Somatic SNVs/InDels</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html#somatic-copy-number-abberations">Somatic copy number abberations</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Personal Cancer Genome Reporter</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>NCGC - functional annotation of somatic DNA variation</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/intro.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="ncgc-functional-annotation-of-somatic-dna-variation">
<h1>NCGC - functional annotation of somatic DNA variation<a class="headerlink" href="#ncgc-functional-annotation-of-somatic-dna-variation" title="Permalink to this headline">¶</a></h1>
<p>The NCGC functional annotation workflow contains a set of software tools
and comprehensive genomic datasets for annotation and interpretation of
single nucleotide variants (SNVs), short insertions/deletions (InDels),
and copy number segments discovered from high-throughput sequencing.
Currently, it builds largely upon <a class="reference external" href="http://www.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor
(VEP)</a> and
<a class="reference external" href="http://github.com/brentp/vcfanno">vcfanno</a>. It has the following
focus:</p>
<ul class="simple">
<li>Cancer relevance</li>
<li>Coding sequence variants (i.e. exome-type studies)</li>
<li><em>insilico</em> predictions of effect of coding variants</li>
<li>Overlap with existing variant databases, both somatic (e.g. hotspots
in different tumor types) and germline</li>
<li>Clinical utility (i.e. clinical variant associations, drug-gene
interactions, drug sensitivity markers etc.)</li>
</ul>
<div class="section" id="data-sources-included">
<h2>Data sources included<a class="headerlink" href="#data-sources-included" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference external" href="http://www.ensembl.org/info/docs/tools/vep/index.html">VEP v85</a> -
Variant Effect Predictor release 85 (GENCODE v19)</li>
<li><a class="reference external" href="http://cancer.sanger.ac.uk/cosmic/">COSMIC v78</a> - catalogue of
somatic mutations in cancer (Sep 2016)</li>
<li><a class="reference external" href="https://sites.google.com/site/jpopgen/dbNSFP">dBNSFP v3.2</a> -
database of non-synonymous functional predictions (March 2016)</li>
<li><a class="reference external" href="https://dcc.icgc.org/">ICGC v22</a> - Somatic mutations discovered
in all ICGC (International Cancer Genomics Consortium) tumor cohorts
(Aug 2016)</li>
<li><a class="reference external" href="http://exac.broadinstitute.org/">ExAC r0.3.1</a> - germline variant
frequencies exome-wide (March 2016)</li>
<li><a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP b147</a> - database of short
genetic variants (April 2016)</li>
<li><a class="reference external" href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">1000Genomes
phase3</a>
- germline variant frequencies genome-wide (May 2013)</li>
<li><a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a> - database of
clinically related variants (Nov 2016)</li>
<li><a class="reference external" href="http://docm.genome.wustl.edu">DoCM</a> - database of curated
mutations (v3.2, April 2016)</li>
<li><a class="reference external" href="http://civic.genome.wustl.edu">CIViC</a> - clinical interpretations
of variants in cancer (Nov 12th 2016)</li>
<li><a class="reference external" href="http://www.cancergenomeinterpreter.org/biomarkers">CBMDB</a> -
Cancer BioMarkers database (Oct 28th 2016)</li>
<li><a class="reference external" href="https://www.intogen.org/downloads">IntOGen catalogs of driver
mutations/genes</a> - (May 2016)</li>
<li><a class="reference external" href="http://cancerhotspots.org">Cancer Hotspots</a> - a resource for
statistically significant mutations in cancer (2016)</li>
<li><a class="reference external" href="http://www.uniprot.org">UniProt/SwissProt KnowledgeBase 2016_09</a>
- resource on protein sequence and functional information (Sep 2016)</li>
<li><a class="reference external" href="http://pfam.xfam.org">Pfam v30</a> - database of protein families
and domains (June 2016)</li>
<li><a class="reference external" href="http://appris.bioinfo.cnio.es/#/">Appris 2016_03.v15</a> -
annotation of prinicipal human transcript isoforms database (March
2016)</li>
<li><a class="reference external" href="http://dgidb.genome.wustl.edu">DGIdb</a> - database of interactions
betweeen antineoplastic drugs and human proteins (v2.22, February
2016)</li>
<li><a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene v2.0</a> - tumor
suppressor/oncogene database (November 2015)</li>
</ul>
<div style="page-break-after: always;"></div></div>
<div class="section" id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h2>
<div class="section" id="somatic-snvs-indels">
<h3>Somatic SNVs/InDels<a class="headerlink" href="#somatic-snvs-indels" title="Permalink to this headline">¶</a></h3>
<div class="section" id="variant-call-format-vcf">
<h4>1. Variant call format - VCF<a class="headerlink" href="#variant-call-format-vcf" title="Permalink to this headline">¶</a></h4>
<p>Each tumor-normal sample pair is provided with an annotated VCF file
with the following naming convention:</p>
<p><em>&lt;sample_pair_identifier&gt;.pass.all.annotated.vcf.gz</em></p>
<p>Following common standards, all VCF files are compressed with
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and indexed with
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a>.</p>
<p>Below follows a description of all annotations/tags present in the VCF
INFO column after processing with the NCGC annotation pipeline:</p>
<ul class="simple">
<li>VEP output</li>
</ul>
<ol class="arabic simple">
<li>CSQ - Complete consequence annotations from VEP. Format:
Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|
EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|
Codons|Existing_variation|ALLELE_NUM|DISTANCE|STRAND|FLAGS|PICK|VARIANT_CLASS|
SYMBOL_SOURCE|HGNC_ID|CANONICAL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|
UNIPARC|RefSeq|DOMAINS|HGVS_OFFSET|CLIN_SIG|SOMATIC|PHENO|MOTIF_NAME|
MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE</li>
</ol>
<ul class="simple">
<li>Consequence - Impact modifier for the consequence type (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Gene - Ensembl stable ID of affected gene (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>Feature_type - Type of feature. Currently one of Transcript,
RegulatoryFeature, MotifFeature (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>Feature - Ensembl stable ID of feature (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>cDNA_position - Relative position of base pair in cDNA sequence
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CDS_position - Relative position of base pair in coding sequence
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CDS_CHANGE - Coding, transcript-specific sequence annotation (picked
by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Protein_position - Relative position of amino acid in protein
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Amino_acids - Only given if the variant affects the protein-coding
sequence (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Codons - The alternative codons with the variant base in upper case
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>IMPACT - Impact modifier for the consequence type (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>VARIANT_CLASS - Sequence Ontology variant class (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>SYMBOL - Gene symbol (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>SYMBOL_SOURCE - The source of the gene symbol (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>STRAND - The DNA strand (1 or -1) on which the transcript/feature
lies (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>ENSP - The Ensembl protein identifier of the affected transcript
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>FLAGS - Transcript quality flags: cds_start_NF: CDS 5&#8217;, incomplete
cds_end_NF: CDS 3&#8217; incomplete (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>SWISSPROT - Best match UniProtKB/Swiss-Prot accession of protein
product (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>TREMBL - Best match UniProtKB/TrEMBL accession of protein product
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>UNIPARC - Best match UniParc accession of protein product (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HGVSc - The HGVS coding sequence name (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>HGVSp - The HGVS protein sequence name (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>HGVSp_short - The HGVS protein sequence name, short version (picked
by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HGVS_OFFSET - Indicates by how many bases the HGVS notations for
this variant have been shifted (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>MOTIF_NAME - The source and identifier of a transcription factor
binding profile aligned at this position (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>MOTIF_POS - The relative position of the variation in the aligned
TFBP (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HIGH_INF_POS - A flag indicating if the variant falls in a high
information position of a transcription factor binding profile (TFBP)
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>MOTIF_SCORE_CHANGE - The difference in motif score of the reference
and variant sequences for the TFBP (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>CELL_TYPE - List of cell types and classifications for regulatory
feature (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CANONICAL - A flag indicating if the transcript is denoted as the
canonical transcript for this gene (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>CCDS - The CCDS identifier for this transcript, where applicable
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>INTRON - The intron number (out of total number) (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>EXON - The exon number (out of total number) (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>DOMAINS - The source and identifier of any overlapping protein
domains (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>DISTANCE - Shortest distance from variant to transcript (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>BIOTYPE - Biotype of transcript or regulatory feature (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>TSL - Transcript support level (picked by VEP&#8217;s &#8211;flag_pick_allele
option)&gt;</li>
<li>PUBMED - PubMed ID(s) of publications that cite existing variant -
VEP</li>
<li>SOMATIC - Somatic status of existing variant(s) - VEP</li>
<li>PHENO - Indicates if existing variant is associated with a phenotype,
disease or trait - VEP</li>
<li>GENE_PHENO - Indicates if overlapped gene is associated with a
phenotype, disease or trait - VEP</li>
<li>ALLELE_NUM - Allele number from input; 0 is reference, 1 is first
alternate etc - VEP</li>
<li>REFSEQ_MATCH - The RefSeq transcript match status; contains a number
of flags indicating whether this RefSeq transcript matches the
underlying reference sequence and/or an Ensembl transcript (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>PICK - Indicates if this block of consequence data was picked by
VEP&#8217;s &#8211;flag_pick_allele option</li>
<li>VEP_ALL_CONSEQUENCE - All transcript consequences
(Consequence:SYMBOL:Feature_type:Feature:BIOTYPE) - VEP</li>
<li><em>Gene information</em></li>
</ul>
<ol class="arabic simple">
<li>ENTREZ_ID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/gene">Entrez</a> gene
identifier</li>
</ol>
<ul class="simple">
<li>APPRIS - Principal isoform flags according to the <a class="reference external" href="http://appris.bioinfo.cnio.es/#/downloads">APPRIS principal
isoform database</a></li>
<li>UNIPROT_ID - <a class="reference external" href="http://www.uniprot.org">UniProt</a> identifier</li>
<li>CANCER_CENSUS_SOMATIC - Gene with known cancer association -
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/census/">Cancer Gene Census,
WTSI</a></li>
<li>CANCER_CENSUS_GERMLINE - Gene with known cancer association -
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/census/">Cancer Gene Census,
WTSI</a></li>
<li>TUMOR_SUPPRESSOR - Gene is predicted as tumor suppressor candidate
according to (<a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene
v2.0</a>)</li>
<li>ONCOGENE - Gene is curated as an oncogene according to (<a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene
v2.0</a>)</li>
<li>ONCOSCORE - Literature-derived score for cancer gene relevance
<a class="reference external" href="http://bioconductor.org/packages/release/bioc/html/OncoScore.html">Bioconductor/OncoScore</a>,
range from 0 (low oncogenic potential) to 1 (high oncogenic
potential)</li>
<li>INTOGEN_DRIVER - Gene is predicted as a cancer driver in the
<a class="reference external" href="https://www.intogen.org/downloads">IntoGen Cancer Drivers Database -
2014.12</a></li>
<li><em>Variant effect and protein-coding information</em></li>
<li>CANCER_MUTATION_HOTSPOT - mutation hotspot codon in
<a class="reference external" href="http://cancerhotspots.org/">cancerhotspots.org</a>. Format:
gene_symbol | codon | q-value</li>
<li>UNIPROT_FEATURE - Overlapping protein annotations from <a class="reference external" href="http://www.uniprot.org">UniProt
KB</a></li>
<li>INTOGEN_DRIVER_MUT - Indicates if existing variant is predicted as
driver mutation from IntoGen Catalog of Driver Mutations</li>
<li>EFFECT_PREDICTIONS - Predictions of effect of variant on protein
function and pre-mRNA splicing from <a class="reference external" href="https://sites.google.com/site/jpopgen/dbNSFP">database of non-synonymous
functional predictions - dbNSFP
v3.2</a>. Predicted
effects are provided by 14 different sources/algorithms (separated by
&#8216;&amp;&#8217;):<ol class="arabic">
<li><a class="reference external" href="http://provean.jcvi.org/index.php">SIFT</a> (Jan 2015)</li>
<li><a class="reference external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen2</a> (v 2.2.2,
predictions based on
<a class="reference external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview#prediction">HumDiv</a>)</li>
<li><a class="reference external" href="http://www.genetics.wustl.edu/jflab/lrt_query.html">LRT</a>
(2009)</li>
<li><a class="reference external" href="http://www.mutationtaster.org/">MutationTaster</a> (data release
Nov 2015)</li>
<li><a class="reference external" href="http://mutationassessor.org/">MutationAssessor</a> (release 3)</li>
<li>[FATHMM] (<a class="reference external" href="http://fathmm.biocompute.org.uk">http://fathmm.biocompute.org.uk</a>) (v2.3)</li>
<li><a class="reference external" href="http://provean.jcvi.org/index.php">PROVEAN</a> (v1.1 Jan 2015)</li>
<li><a class="reference external" href="http://fathmm.biocompute.org.uk/fathmmMKL.htm">FATHMM_MKL</a></li>
<li><a class="reference external" href="http://cadd.gs.washington.edu/">CADD</a> (v1.3)</li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/25552646">DBNSFP_CONSENSUS_SVM</a>
(Ensembl/consensus prediction, based on support vector machines)</li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/25552646">DBNSFP_CONSENSUS_LR</a>
(Ensembl/consensus prediction, logistic regression based)</li>
<li><a class="reference external" href="http://nar.oxfordjournals.org/content/42/22/13534">SPLICE_SITE_EFFECT_ADA</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on
adaptive boosting)</li>
<li><a class="reference external" href="http://nar.oxfordjournals.org/content/42/22/13534">SPLICE_SITE_EFFECT_RF</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on
adaptive boosting)</li>
<li><a class="reference external" href="http://nar.oxfordjournals.org/content/42/22/13534">SPLICE_SITE_EFFECT_BOOL</a>
(predicts &#8216;affected_splicing&#8217; if SPLICE_SITE_EFFECT_ADA or
SPLICE_SITE_EFFECT_RF predicts &#8216;affected_splicing&#8217;</li>
</ol>
</li>
<li><em>Variant overlap and frequencies in germline/somatic databases</em></li>
</ul>
<ol class="arabic simple">
<li>AFR_AF_EXAC - African/American germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
0.3.1</a>)</li>
</ol>
<ul class="simple">
<li>AMR_AF_EXAC - American germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
0.3.1</a>)</li>
<li>GLOBAL_AF_EXAC - Adjusted global germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
0.3.1</a>)</li>
<li>EAS_AF_EXAC - East Asian germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
0.3.1</a>)</li>
<li>FIN_AF_EXAC - Finnish germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation
Consortium release 0.3.1</a>)</li>
<li>NFE_AF_EXAC - Non-Finnish European germline allele frequency
(<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation Consortium release
0.3.1</a>)</li>
<li>OTH_AF_EXAC - Other germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation
Consortium release 0.3.1</a>)</li>
<li>SAS_AF_EXAC - South Asian germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
0.3</a>)</li>
<li>AFR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from AFR (African)</li>
<li>AMR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from AMR (Ad Mixed American)</li>
<li>EAS_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from EAS (East Asian)</li>
<li>EUR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from EUR (European)</li>
<li>SAS_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from SAS (South Asian)</li>
<li>GLOBAL_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for all
1000G project samples (global)</li>
<li>DBSNPRSID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a> reference ID</li>
<li>DBSNPBUILDID - Initial <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a>
build ID for rsID</li>
<li>DBSNP_MAPPINGSTATUS - Status with respect to the genomic mappability
of the flanking sequence of the rsID</li>
<li>DBSNP_VALIDATION - Categories of evidence that support the variant
in <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a></li>
<li>DBSNP_SUBMISSIONS - Number of individual submissions to rsID</li>
<li>GWAS_CATALOG_PMID - Variant is linked to phenotype through the
<a class="reference external" href="https://www.ebi.ac.uk/gwas/">GWAS Catalog</a>, literature in PMID
list</li>
<li>GWAS_CATALOG_TRAIT_URI - List of trait URIs for GWAS-associated
variant</li>
<li>COSMIC_MUTATION_ID - Mutation identifier in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in cancer - COSMIC
v78</a>
database</li>
<li>COSMIC_CODON_FRAC_GW - For different tumor types, number of
samples mutated at associated codon position (format:
codon_number:tumor_type:fraction_mutated). Samples subject to
exome/genome-wide screens only <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer - COSMIC
v78</a>.</li>
<li>COSMIC_CODON_COUNT_GW - For different tumor types, number of
samples mutated at associated codon position (format:
codon_number:tumor_type:frequency). Samples subject to
exome/genome-wide screens only <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer - COSMIC
v78</a></li>
<li>COSMIC_COUNT_GW - Global frequency of variant in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of
somatic mutations in cancer - COSMIC
v78</a>.</li>
<li>COSMIC_SITE_HISTOLOGY - Primary site/histology distribution across
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in cancer - COSMIC
v78</a>.</li>
<li>COSMIC_CANCER_TYPE_GW - Frequency of variant across different
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in cancer - COSMIC
v78</a> -
samples subject to exome/genome-wide screens only</li>
<li>COSMIC_CANCER_TYPE_ALL - Frequency of variant across different
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in cancer - COSMIC
v78</a></li>
<li>COSMIC_SAMPLE_SOURCE - Sample source distribution for variant in
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in cancer - COSMIC
v78</a>.</li>
<li>COSMIC_DRUG_RESISTANCE - Targeted drugs/therapies subject to
resistance in tumors that carry the mutation. <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in cancer - COSMIC
v78</a>.</li>
<li>COSMIC_FATHMM_PRED - Variant effect prediction from COSMIC&#8217;s FATHMM
algorithm (COSMIC variants only) <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer - COSMIC
v78</a>.</li>
<li>COSMIC_VARTYPE - COSMIC variant type <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations
in cancer - COSMIC
v78</a>.</li>
<li>COSMIC_CONSEQUENCE - COSMIC consequence type <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in cancer - COSMIC
v78</a>.</li>
<li>ICGC_PROJECTS - Variant frequency count in different <a class="reference external" href="https://dcc.icgc.org/repository/current/Projects">ICGC Project
IDs</a></li>
<li><em>Clinical variant associations</em></li>
</ul>
<ol class="arabic simple">
<li>CLINVAR_MSID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a>
Measure Set/Variant ID</li>
<li>CLINVAR_PMIDS - Associated Pubmed IDs for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>CLINVAR_SIG - Clinical significance for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>CLINVAR_VARIANT_ORIGIN - Origin of variant (somatic, germline, de
novo etc.) for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>DOCM_DISEASE - Associated disease types for variant in <a class="reference external" href="http://docm.genome.wustl.edu">Database of
Curated Mutations</a></li>
<li>DOCM_PMID - Associated Pubmed IDs for variant in <a class="reference external" href="http://docm.genome.wustl.edu">Database of
Curated Mutations</a></li>
</ol>
<ul class="simple">
<li><em>Other</em><ol class="arabic">
<li>VT - Variant type (snv, INdel, inDEL, block_substitution)</li>
</ol>
</li>
<li>ANTINEOPLASTIC_DRUG_INTERACTION - Approved and experimental
antineoplastic drugs interacting with the mutated gene, as retrieved
from the <a class="reference external" href="http://dgidb.genome.wustl.edu/">Drug-Gene Interaction
Database</a></li>
<li>CIVIC_ID, CIVIC_ID_2 - Variant identifiers in the <a class="reference external" href="http://civic.genome.wustl.edu">CIViC
database</a></li>
<li>CBMDB_ID - Variant identifier in the <a class="reference external" href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer bioMarkers
database</a></li>
<li>CALL_CONFIDENCE - STRONG, WEAK_MUTECT, WEAK_STRELKA</li>
</ul>
</div>
<div class="section" id="tab-separated-values-tsv">
<h4>2. Tab-separated values (TSV)<a class="headerlink" href="#tab-separated-values-tsv" title="Permalink to this headline">¶</a></h4>
<div class="section" id="annotated-list-of-all-snvs-indels">
<h5>Annotated List of all SNVs/InDEls<a class="headerlink" href="#annotated-list-of-all-snvs-indels" title="Permalink to this headline">¶</a></h5>
<p>We provide a tab-separated values file with most important annotations
for SNVs/InDels. The file has the following naming convention:</p>
<p><em>&lt;sample_pair_identifier&gt;.snvs_indels.tiers.tsv</em></p>
<p>The SNVs/InDels are organized into different <strong>tiers</strong> that reflect
relevance for therapeutics/tumorigenesis:</p>
<ul class="simple">
<li><strong>Tier 1</strong> constitute variants recorded as prognostic/diagnostic/drug
sensitivity biomarkers in the <a class="reference external" href="http://civic.genome.wustl.edu">CIViC
database</a> and the <a class="reference external" href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer
Biomarkers
Database</a></li>
<li><strong>Tier 2</strong> includes other coding variants that are found in known
mutational hotspots, predicted as cancer driver mutations, or curated
as disease-causing</li>
<li><strong>Tier 3</strong> includes other coding variants found in oncogenes, tumor
suppressor genes, or cancer census genes</li>
<li><strong>Tier 4</strong> includes other coding variants</li>
<li><strong>Tier 5</strong> includes non-coding variants</li>
</ul>
<p><strong>Note</strong>: &#8216;<em>coding variants</em>&#8216; refer to the set of variants with the
following consequences: - missense variant - splice donor/splice
acceptor alteration - stop gained/stop lost - frameshift/non-frameshift
variants</p>
<p>The following variables are included in the tiered TSV file:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">GENOMIC_CHANGE</span> <span class="o">-</span> <span class="n">Identifier</span> <span class="k">for</span> <span class="n">genomic</span> <span class="n">variant</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">g</span><span class="o">.</span><span class="n">chr1</span><span class="p">:</span><span class="mi">152382569</span><span class="p">:</span><span class="n">A</span><span class="o">&gt;</span><span class="n">G</span>
<span class="mf">2.</span> <span class="n">GENOME_VERSION</span> <span class="o">-</span> <span class="n">Assembly</span> <span class="n">version</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">GRCh37</span>
<span class="mf">3.</span> <span class="n">VCF_SAMPLE_ID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
<span class="mf">4.</span> <span class="n">VARIANT_CLASS</span> <span class="o">-</span> <span class="n">Variant</span> <span class="nb">type</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">SNV</span><span class="o">/</span><span class="n">insertion</span><span class="o">/</span><span class="n">deletion</span>
<span class="mf">5.</span> <span class="n">SYMBOL</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">symbol</span>
<span class="mf">6.</span> <span class="n">GENE_NAME</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">description</span>
<span class="mf">7.</span> <span class="n">CCDS</span> <span class="o">-</span> <span class="n">CCDS</span> <span class="n">identifier</span>
<span class="mf">8.</span> <span class="n">ENTREZ_ID</span> <span class="o">-</span> <span class="n">Entrez</span> <span class="n">gene</span> <span class="n">identifier</span>
<span class="mf">9.</span> <span class="n">UNIPROT_ID</span> <span class="o">-</span> <span class="n">UniProt</span> <span class="n">protein</span> <span class="n">identifier</span>
<span class="mf">10.</span> <span class="n">ONCOSCORE</span> <span class="o">-</span> <span class="n">Literature</span><span class="o">-</span><span class="n">derived</span> <span class="n">score</span> <span class="k">for</span> <span class="n">cancer</span> <span class="n">gene</span> <span class="n">relevance</span>
<span class="mf">11.</span> <span class="n">ONCOGENE</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">curated</span> <span class="k">as</span> <span class="n">an</span> <span class="n">oncogene</span> <span class="n">according</span> <span class="n">to</span> <span class="n">TSGene</span>
<span class="mf">12.</span> <span class="n">TUMOR_SUPPRESSOR</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span> <span class="n">tumor</span> <span class="n">suppressor</span>
    <span class="n">candidate</span> <span class="n">according</span> <span class="n">to</span> <span class="n">TSGene</span>
<span class="mf">13.</span> <span class="n">INTOGEN_DRIVER</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span> <span class="n">a</span> <span class="n">cancer</span> <span class="n">driver</span> <span class="ow">in</span> <span class="n">the</span>
    <span class="n">IntoGen</span> <span class="n">Cancer</span> <span class="n">Drivers</span> <span class="n">Database</span> <span class="o">-</span> <span class="mf">2014.12</span>
<span class="mf">14.</span> <span class="n">CANCER_CENSUS_SOMATIC</span> <span class="o">-</span> <span class="n">Gene</span> <span class="k">with</span> <span class="n">known</span> <span class="n">cancer</span> <span class="n">association</span> <span class="o">-</span>
    <span class="n">Cancer</span> <span class="n">Gene</span> <span class="n">Census</span><span class="p">,</span> <span class="n">WTSI</span>
<span class="mf">15.</span> <span class="n">CANCER_CENSUS_GERMLINE</span> <span class="o">-</span> <span class="n">Gene</span> <span class="k">with</span> <span class="n">known</span> <span class="n">cancer</span> <span class="n">association</span> <span class="o">-</span>
    <span class="n">Cancer</span> <span class="n">Gene</span> <span class="n">Census</span><span class="p">,</span> <span class="n">WTSI</span>
<span class="mf">16.</span> <span class="n">CONSEQUENCE</span> <span class="o">-</span> <span class="n">Variant</span> <span class="n">consequence</span> <span class="p">(</span><span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span> <span class="n">output</span><span class="p">:</span>
    <span class="n">Consequence</span><span class="p">)</span>
<span class="mf">17.</span> <span class="n">PROTEIN_CHANGE</span> <span class="o">-</span> <span class="n">Protein</span> <span class="n">change</span> <span class="p">(</span><span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span> <span class="n">output</span><span class="p">:</span>
    <span class="n">HGVSp_short</span><span class="p">)</span>
<span class="mf">18.</span> <span class="n">PROTEIN_DOMAIN</span> <span class="o">-</span> <span class="n">Protein</span> <span class="n">domain</span>
<span class="mf">19.</span> <span class="n">CDS_CHANGE</span> <span class="o">-</span> <span class="n">composite</span> <span class="n">variable</span> <span class="k">for</span> <span class="n">coding</span> <span class="n">change</span><span class="p">,</span> <span class="nb">format</span><span class="p">:</span>
    <span class="n">Consequence</span><span class="p">:</span><span class="n">Feature</span><span class="p">:</span><span class="n">cDNA_position</span><span class="p">:</span><span class="n">EXON</span><span class="p">:</span><span class="n">HGVSp_short</span>
<span class="mf">20.</span> <span class="n">EFFECT_PREDICTIONS</span> <span class="o">-</span> <span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span>
<span class="mf">21.</span> <span class="n">CANCER_MUTATION_HOTSPOT</span> <span class="o">-</span> <span class="n">mutation</span> <span class="n">hotspot</span> <span class="n">codon</span> <span class="ow">in</span>
    <span class="n">cancerhotspots</span><span class="o">.</span><span class="n">org</span><span class="o">.</span> <span class="n">Format</span><span class="p">:</span> <span class="n">gene_symbol</span> <span class="o">|</span> <span class="n">codon</span> <span class="o">|</span> <span class="n">q</span><span class="o">-</span><span class="n">value</span>
<span class="mf">22.</span> <span class="n">INTOGEN_DRIVER_MUT</span> <span class="o">-</span> <span class="n">Indicates</span> <span class="k">if</span> <span class="n">existing</span> <span class="n">variant</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span>
    <span class="n">driver</span> <span class="n">mutation</span> <span class="kn">from</span> <span class="nn">IntoGen</span> <span class="n">Catalog</span> <span class="n">of</span> <span class="n">Driver</span> <span class="n">Mutations</span>
<span class="mf">23.</span> <span class="n">VEP_ALL_CONSEQUENCE</span> <span class="o">-</span> <span class="nb">all</span> <span class="n">VEP</span> <span class="n">consequences</span>
<span class="mf">24.</span> <span class="n">DBSNP</span> <span class="o">-</span> <span class="n">dbSNP</span> <span class="n">reference</span> <span class="n">cluster</span> <span class="n">ID</span>
<span class="mf">25.</span> <span class="n">COSMIC</span> <span class="o">-</span> <span class="n">COSMIC</span> <span class="n">mutation</span> <span class="n">ID</span>
<span class="mf">26.</span> <span class="n">COSMIC_SITE_HISTOLOGY</span> <span class="o">-</span> <span class="n">distribution</span> <span class="n">of</span> <span class="n">tumor</span> <span class="n">sites</span><span class="o">/</span><span class="n">histology</span> <span class="n">types</span>
    <span class="k">for</span> <span class="n">COSMIC</span> <span class="n">mutation</span>
<span class="mf">27.</span> <span class="n">COSMIC_DRUG_RESISTANCE</span> <span class="o">-</span> <span class="n">variant</span> <span class="n">associated</span> <span class="k">with</span> <span class="n">resistance</span> <span class="n">to</span> <span class="n">a</span>
    <span class="n">particular</span> <span class="n">antineoplastic</span> <span class="n">drug</span>
<span class="mf">28.</span> <span class="n">CLINVAR</span> <span class="o">-</span> <span class="n">variant</span> <span class="n">origin</span> <span class="ow">and</span> <span class="n">associated</span> <span class="n">traits</span> <span class="n">associated</span> <span class="k">with</span> <span class="n">variant</span>
<span class="mf">29.</span> <span class="n">CLINVAR_SIG</span> <span class="o">-</span> <span class="n">clinical</span> <span class="n">significance</span> <span class="n">of</span> <span class="n">CLINVAR</span> <span class="n">variant</span>
<span class="mf">30.</span> <span class="n">GLOBAL_AF_EXAC</span> <span class="o">-</span> <span class="n">adjusted</span> <span class="k">global</span> <span class="n">germline</span> <span class="n">allele</span> <span class="n">frequency</span> <span class="ow">in</span> <span class="n">ExAC</span> <span class="n">release</span> <span class="mf">0.3</span><span class="o">.</span><span class="mi">1</span>
<span class="mf">31.</span> <span class="n">GLOBAL_AF_1KG</span> <span class="o">-</span> <span class="mi">1000</span><span class="n">G</span> <span class="n">Project</span> <span class="o">-</span> <span class="n">phase</span> <span class="mi">3</span><span class="p">,</span> <span class="n">germline</span> <span class="n">allele</span> <span class="n">frequency</span>
    <span class="k">for</span> <span class="nb">all</span> <span class="mi">1000</span><span class="n">G</span> <span class="n">project</span> <span class="n">samples</span> <span class="p">(</span><span class="k">global</span><span class="p">)</span>
<span class="mf">32.</span> <span class="n">CALL_CONFIDENCE</span> <span class="o">-</span> <span class="n">_STRONG</span> <span class="o">/</span> <span class="n">WEAK_MUTECT</span> <span class="o">/</span> <span class="n">WEAK_STRELKA</span>
<span class="mf">33.</span> <span class="n">DP_TUMOR</span> <span class="o">-</span> <span class="n">tumor</span> <span class="n">depth</span>
<span class="mf">34.</span> <span class="n">AF_TUMOR</span> <span class="o">-</span> <span class="n">allelic</span> <span class="n">fraction</span> <span class="n">tumor</span>
<span class="mf">35.</span> <span class="n">DP_NORMAL</span> <span class="o">-</span> <span class="n">normal</span> <span class="n">depth</span>
<span class="mf">36.</span> <span class="n">AF_NORMAL</span> <span class="o">-</span> <span class="n">allelic</span> <span class="n">fraction</span> <span class="n">normal</span>
<span class="mf">37.</span> <span class="n">TIER</span>
<span class="mf">38.</span> <span class="n">TIER_DESCRIPTION</span>
</pre></div>
</div>
</div>
<div class="section" id="biomarkers-among-snvs-indels">
<h5>Biomarkers among SNVs/InDEls<a class="headerlink" href="#biomarkers-among-snvs-indels" title="Permalink to this headline">¶</a></h5>
<p>For tumor samples that have hits in <strong>Tier 1</strong> we provide an additional
file with all underlying clinical evidence items. The file has the
following naming convention:</p>
<p><em>&lt;sample_pair_identifier&gt;.snvs_indels.biomarkers.tsv</em></p>
<p>The format of the biomarker TSV file is as follows:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">GENOMIC_CHANGE</span> <span class="o">-</span> <span class="n">Identifier</span> <span class="k">for</span> <span class="n">genomic</span> <span class="n">variant</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">g</span><span class="o">.</span><span class="n">chr1</span><span class="p">:</span><span class="mi">152382569</span><span class="p">:</span><span class="n">A</span><span class="o">&gt;</span><span class="n">G</span>
<span class="mf">2.</span> <span class="n">GENOME_VERSION</span> <span class="o">-</span> <span class="n">Assembly</span> <span class="n">version</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">GRCh37</span>
<span class="mf">3.</span> <span class="n">VCF_SAMPLE_ID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
<span class="mf">4.</span> <span class="n">SYMBOL</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">symbol</span>
<span class="mf">5.</span> <span class="n">CONSEQUENCE</span> <span class="o">-</span> <span class="n">Variant</span> <span class="n">consequence</span>
<span class="mf">6.</span> <span class="n">BM_CLINICAL_SIGNIFICANCE</span> <span class="o">-</span> <span class="n">The</span> <span class="n">association</span> <span class="k">with</span> <span class="n">diagnostic</span><span class="o">/</span><span class="n">prognostic</span> <span class="n">end</span> <span class="n">point</span> <span class="ow">or</span> <span class="n">treatment</span>
<span class="mf">7.</span> <span class="n">BM_EVIDENCE_LEVEL</span> <span class="o">-</span> <span class="n">The</span> <span class="nb">type</span> <span class="n">of</span> <span class="n">experiment</span> <span class="kn">from</span> <span class="nn">which</span> <span class="n">the</span> <span class="n">evidence</span> <span class="ow">is</span> <span class="n">curated</span>
                <span class="p">(</span><span class="n">validated</span><span class="p">,</span> <span class="n">clinical</span><span class="p">,</span> <span class="n">pre</span><span class="o">-</span><span class="n">clinical</span><span class="p">,</span> <span class="n">case</span> <span class="n">study</span><span class="p">,</span> <span class="ow">and</span> <span class="n">inferential</span><span class="p">)</span>
<span class="mf">8.</span> <span class="n">BM_EVIDENCE_TYPE</span> <span class="o">-</span> <span class="n">Category</span> <span class="n">of</span> <span class="n">clinical</span> <span class="n">action</span><span class="o">/</span><span class="n">relevance</span> <span class="n">implicated</span> <span class="n">by</span> <span class="n">event</span>
                <span class="p">(</span><span class="n">Predictive</span><span class="p">,</span> <span class="n">Prognostic</span><span class="p">,</span> <span class="n">Predisposing</span> <span class="ow">and</span> <span class="n">Diagnostic</span><span class="p">)</span>
<span class="mf">9.</span> <span class="n">BM_EVIDENCE_DIRECTION</span> <span class="o">-</span> <span class="n">An</span> <span class="n">indicator</span> <span class="n">of</span> <span class="n">whether</span> <span class="n">the</span> <span class="n">evidence</span> <span class="n">statement</span> <span class="n">supports</span> <span class="ow">or</span>
                                <span class="n">refutes</span> <span class="n">the</span> <span class="n">clinical</span> <span class="n">significance</span> <span class="n">of</span> <span class="n">an</span> <span class="n">event</span>
<span class="mf">10.</span> <span class="n">BM_DISEASE_NAME</span> <span class="o">-</span> <span class="n">Specific</span> <span class="n">disease</span> <span class="ow">or</span> <span class="n">disease</span> <span class="n">subtype</span> <span class="n">that</span> <span class="ow">is</span> <span class="n">associated</span> <span class="k">with</span>
                            <span class="n">this</span> <span class="n">event</span> <span class="ow">and</span> <span class="n">its</span> <span class="n">clinical</span> <span class="n">implication</span>
<span class="mf">11.</span> <span class="n">BM_DRUG_NAMES</span> <span class="o">-</span> <span class="n">For</span> <span class="n">predictive</span> <span class="n">evidence</span><span class="p">,</span> <span class="n">indicates</span> <span class="n">the</span> <span class="n">therapy</span> <span class="k">for</span> <span class="n">which</span> <span class="n">sensitivity</span>
                        <span class="ow">or</span> <span class="n">resistance</span> <span class="ow">is</span> <span class="n">indicated</span>
<span class="mf">12.</span> <span class="n">BM_RATING</span> <span class="o">-</span> <span class="n">A</span> <span class="n">rating</span> <span class="n">on</span> <span class="n">a</span> <span class="mi">5</span><span class="o">-</span><span class="n">star</span> <span class="n">scale</span><span class="p">,</span> <span class="n">portraying</span> <span class="n">the</span> <span class="n">curators</span> <span class="n">trust</span> <span class="ow">in</span> <span class="n">the</span>
                    <span class="n">experiments</span> <span class="kn">from</span> <span class="nn">which</span> <span class="n">the</span> <span class="n">evidence</span> <span class="ow">is</span> <span class="n">curated</span>
<span class="mf">13.</span> <span class="n">BM_CITATION</span> <span class="o">-</span> <span class="n">Publication</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="n">where</span> <span class="n">the</span> <span class="n">event</span> <span class="n">was</span> <span class="n">described</span><span class="o">/</span><span class="n">explored</span><span class="o">/</span><span class="n">guidelines</span><span class="o">/</span><span class="n">trials</span>
<span class="mf">14.</span> <span class="n">TIER</span>
<span class="mf">15.</span> <span class="n">TIER_DESCRIPTION</span>
</pre></div>
</div>
</div>
<div class="section" id="mutational-signatures">
<h5>Mutational signatures<a class="headerlink" href="#mutational-signatures" title="Permalink to this headline">¶</a></h5>
<p>For each tumor sample, we apply the <a class="reference external" href="https://github.com/raerose01/deconstructSigs">deconstructSigs
package</a> to delineate
the known mutational signatures. The inferred, weighted contributions by
each signature and their underlying, proposed etiologies are given in a
TSV file with the following naming convention:</p>
<p><em>&lt;sample_pair_identifier&gt;.mutational_signatures.tsv</em></p>
<p>The format of the mutational sigantures TSV file is as follows:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">Signature_ID</span> <span class="o">-</span> <span class="n">ID</span> <span class="n">of</span> <span class="n">signature</span> <span class="kn">from</span> <span class="nn">COSMIC</span><span class="s1">&#39;s 30 reference signatures</span>
<span class="mf">2.</span> <span class="n">Weight</span> <span class="o">-</span> <span class="n">inferred</span> <span class="n">weight</span> <span class="n">of</span> <span class="n">signature</span> <span class="ow">in</span> <span class="n">the</span> <span class="n">tumor</span> <span class="n">sample</span>
<span class="mf">3.</span> <span class="n">Cancer_types</span> <span class="o">-</span> <span class="n">cancer</span> <span class="n">types</span> <span class="ow">in</span> <span class="n">which</span> <span class="n">the</span> <span class="n">signature</span> <span class="n">has</span> <span class="n">been</span> <span class="n">observed</span>
<span class="mf">4.</span> <span class="n">Proposed_aetiology</span> <span class="o">-</span> <span class="n">proposed</span> <span class="n">underlying</span> <span class="n">etiology</span>
<span class="mf">5.</span> <span class="n">SampleID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="somatic-copy-number-abberations">
<h3>Somatic copy number abberations<a class="headerlink" href="#somatic-copy-number-abberations" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id1">
<h4>1. Tab-separated values (TSV)<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<p>Each tumor-normal sample pair is provided with a gzipped, tab-separated
values (TSV) file of somatic copy number segments determined by
<a class="reference external" href="https://sites.google.com/site/mskfacets/">FACETS</a>. Copy number
segments have been intersected with the genomic coordinates of all
transcripts from ENSEMBL/GENCODE. In adddition, we attach
cancer-relevant annotations for the affected transcripts. The naming
convention of the compressed TSV file is as follows:</p>
<p><em>&lt;sample_pair_identifier&gt;.facets.cnv.annotated.gz</em></p>
<p>The format of the TSV file is the following:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span><span class="n">chrom</span> <span class="o">-</span> <span class="n">chromosome</span>
<span class="mf">2.</span><span class="n">segment_start</span> <span class="o">-</span> <span class="n">start</span> <span class="n">of</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">segment</span> <span class="n">determined</span> <span class="n">by</span> <span class="n">FACETS</span>
<span class="mf">3.</span><span class="n">segment_end</span> <span class="o">-</span> <span class="n">end</span> <span class="n">of</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">segment</span> <span class="n">determined</span> <span class="n">by</span> <span class="n">FACETS</span>
<span class="mf">4.</span><span class="n">segment_length</span> <span class="o">-</span> <span class="n">length</span> <span class="n">of</span> <span class="n">segment</span> <span class="ow">in</span> <span class="n">Mb</span>
<span class="mf">5.</span><span class="n">cf</span> <span class="o">-</span> <span class="n">estimated</span> <span class="n">cellular</span> <span class="n">fraction</span>
<span class="mf">6.</span><span class="n">LogR</span> <span class="o">-</span> <span class="n">Copy</span> <span class="n">log</span><span class="o">-</span><span class="n">ratio</span>
<span class="mf">7.</span><span class="n">cnTotal</span> <span class="o">-</span> <span class="n">Estimated</span> <span class="n">total</span> <span class="n">copy</span> <span class="n">number</span>
<span class="mf">8.</span><span class="n">cnMinor</span> <span class="o">-</span> <span class="n">Estimated</span> <span class="n">minor</span> <span class="n">copy</span> <span class="n">number</span>
<span class="mf">9.</span><span class="n">ensembl_gene_id</span>
<span class="mf">10.</span><span class="n">symbol</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">symbol</span>
<span class="mf">11.</span><span class="n">ensembl_transcript_id</span>
<span class="mf">12.</span><span class="n">transcript_start</span>
<span class="mf">13.</span><span class="n">transcript_end</span>
<span class="mf">14.</span><span class="n">transcript_overlap_percent</span> <span class="o">-</span> <span class="n">percent</span> <span class="n">of</span> <span class="n">transcript</span> <span class="n">length</span> <span class="n">covered</span> <span class="n">by</span> <span class="n">CN</span> <span class="n">segment</span>
<span class="mf">15.</span><span class="n">name</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">name</span> <span class="n">description</span>
<span class="mf">16.</span><span class="n">gene_biotype</span> <span class="o">-</span> <span class="nb">type</span> <span class="n">of</span> <span class="n">gene</span>
<span class="mf">17.</span><span class="n">cancer_census_germline</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">implicated</span> <span class="k">with</span> <span class="n">germline</span>
   <span class="n">predisposition</span> <span class="n">to</span> <span class="n">various</span> <span class="n">cancer</span> <span class="n">subtypes</span>
<span class="mf">18.</span><span class="n">cancer_census_somatic</span> <span class="o">-</span> <span class="n">gene</span> <span class="k">for</span> <span class="n">which</span> <span class="n">somatic</span> <span class="n">mutations</span>
   <span class="n">have</span> <span class="n">been</span> <span class="n">causally</span> <span class="n">implicated</span> <span class="ow">in</span> <span class="n">tumor</span> <span class="n">development</span>
<span class="mf">19.</span><span class="n">tsgene</span> <span class="o">-</span> <span class="n">tumor</span> <span class="n">suppressor</span> <span class="n">gene</span> <span class="n">status</span> <span class="p">(</span><span class="n">TSgene</span> <span class="mf">2.0</span> <span class="n">database</span><span class="p">)</span>
<span class="mf">20.</span><span class="n">tsgene_oncogene</span> <span class="o">-</span> <span class="n">oncogene</span> <span class="n">status</span> <span class="p">(</span><span class="n">TSgene</span> <span class="mf">2.0</span> <span class="n">database</span><span class="p">)</span>
<span class="mf">21.</span><span class="n">intogen_drivers</span> <span class="o">-</span> <span class="n">predicted</span> <span class="n">driver</span> <span class="n">gene</span> <span class="n">status</span> <span class="p">(</span><span class="n">IntoGen</span> <span class="n">Cancer</span> <span class="n">Drivers</span> <span class="n">Database</span><span class="p">)</span>
<span class="mf">22.</span><span class="n">antineoplastic_drugs_dgidb</span> <span class="o">-</span> <span class="n">validated</span> <span class="ow">and</span> <span class="n">experimental</span>
   <span class="n">antineoplastic</span> <span class="n">drugs</span> <span class="n">interacting</span> <span class="k">with</span> <span class="n">gene</span>
<span class="mf">23.</span><span class="n">gencode_v19</span> <span class="o">-</span> <span class="n">transcript</span> <span class="ow">is</span> <span class="n">part</span> <span class="n">of</span> <span class="n">GENCODE</span> <span class="n">V19</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>