

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Input &amp; output &mdash; Personal Cancer Genome Reporter 0.1 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="author" title="About these documents"
              href="about.html"/>
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Personal Cancer Genome Reporter 0.1 documentation" href="index.html"/>
        <link rel="prev" title="Annotation resources" href="annotation_resources.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Personal Cancer Genome Reporter
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">Annotation resources</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Input &amp; output</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#vcf">VCF</a></li>
<li class="toctree-l3"><a class="reference internal" href="#copy-number-segments">Copy number segments</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output-interactive-html-report">Output - Interactive HTML report</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-somatic-snvs-indels">Output - Somatic SNVs/InDels</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#variant-call-format-vcf">Variant call format - VCF</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#vep-consequence-annotations"><em>VEP consequence annotations</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#gene-information"><em>Gene information</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#variant-effect-and-protein-coding-information"><em>Variant effect and protein-coding information</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#variant-frequencies-annotations-in-germline-somatic-databases"><em>Variant frequencies/annotations in germline/somatic databases</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#clinical-associations"><em>Clinical associations</em></a></li>
<li class="toctree-l4"><a class="reference internal" href="#other"><em>Other</em></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#tab-separated-values-tsv">Tab-separated values (TSV)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#annotated-list-of-all-snvs-indels">Annotated List of all SNVs/InDels</a></li>
<li class="toctree-l4"><a class="reference internal" href="#biomarkers-among-snvs-indels">Biomarkers among SNVs/InDEls</a></li>
<li class="toctree-l4"><a class="reference internal" href="#mutational-signatures">Mutational signatures</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output-somatic-copy-number-abberations">Output - Somatic copy number abberations</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id1">1. Tab-separated values (TSV)</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Personal Cancer Genome Reporter</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Input &amp; output</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/output.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="input-output">
<h1>Input &amp; output<a class="headerlink" href="#input-output" title="Permalink to this headline">¶</a></h1>
<div class="section" id="input">
<h2>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h2>
<p>The PCGR workflow accepts two types of input files:</p>
<ul class="simple">
<li>An unannotated, single-sample VCF file (&gt;= v4.2) with called somatic
variants (SNVs/InDels)</li>
<li>A copy number segment file</li>
</ul>
<p>PCGR can be run with either or both of the two input files present.</p>
<p><strong>IMPORTANT NOTE</strong>: Only the GRCh37 version of the human genome is
currently supported.</p>
<div class="section" id="vcf">
<h3>VCF<a class="headerlink" href="#vcf" title="Permalink to this headline">¶</a></h3>
<p>The following requirements <strong>MUST</strong> be met by the input VCF for PCGR to
work properly:</p>
<ol class="arabic simple">
<li>Variants in the raw VCF that contain multiple alternative alleles
(e.g. &#8220;multiple ALTs&#8221;) must be split into variants with a single
alternative allele. This can be done with the help of either <a class="reference external" href="http://genome.sph.umich.edu/wiki/Vt#Decompose">vt
decompose</a> or
<a class="reference external" href="https://github.com/vcflib/vcflib#vcflib">vcflib&#8217;s vcfbreakmulti</a>.
We will add integrated support for this in an upcoming release</li>
<li>The contents of the VCF must be sorted correctly (i.e. according to
chromosomal order and chromosomal position). This can be obtained by
<a class="reference external" href="https://vcftools.github.io/perl_module.html#vcf-sort">vcftools</a>.<ul>
<li>We <strong>strongly</strong> recommend that the input VCF is compressed and
indexed using <a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a></li>
<li>&#8216;chr&#8217; must be stripped from the chromosome names</li>
</ul>
</li>
</ol>
<p><strong>IMPORTANT NOTE</strong>: Considering the VCF output for the <a class="reference external" href="https://www.biostars.org/p/19104/">numerous somatic
SNV/InDel callers</a> that have been
developed, we have a experienced a general lack of uniformity and
robustness for the representation of somatic variant genotype data (e.g.
variant allelic depths (tumor/normal), genotype quality etc.). In the
output results provided within the current version of PCGR, we are
considering PASSed variants only, and variant genotype data (i.e. as
found in the VCF SAMPLE columns) are not handled or parsed. As improved
standards for this matter may emerge, we will strive to include this
information in the annotated output files.</p>
</div>
<div class="section" id="copy-number-segments">
<h3>Copy number segments<a class="headerlink" href="#copy-number-segments" title="Permalink to this headline">¶</a></h3>
<p>The tab-separated values file with copy number aberrations <strong>MUST</strong>
contain the following four columns:</p>
<ul class="simple">
<li>Chromosome</li>
<li>Start</li>
<li>End</li>
<li>Segment_Mean</li>
</ul>
<p>Here, <em>Chromosome</em>, <em>Start</em>, and <em>End</em> denote the chromosomal segment
(GRCh37), and <strong>Segment_Mean</strong> denotes the log(2) ratio for a
particular segment, which is a common output of somatic copy number
alteration callers. Below shows the initial part of a copy number
segment file that is formatted correctly according to PCGR&#8217;s
requirements:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">Chromosome</span>    <span class="n">Start</span>   <span class="n">End</span> <span class="n">Segment_Mean</span>
<span class="mi">1</span> <span class="mi">3218329</span> <span class="mi">3550598</span> <span class="mf">0.0024</span>
<span class="mi">1</span> <span class="mi">3552451</span> <span class="mi">4593614</span> <span class="mf">0.1995</span>
<span class="mi">1</span> <span class="mi">4593663</span> <span class="mi">6433129</span> <span class="o">-</span><span class="mf">1.0277</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="output-interactive-html-report">
<h2>Output - Interactive HTML report<a class="headerlink" href="#output-interactive-html-report" title="Permalink to this headline">¶</a></h2>
<p>An interactive and tier-structured HTML report that shows the most
relevant findings in the query cancer genome is provided with the
following naming convention:</p>
<p><strong>sample_id</strong>.pcgr.html</p>
<p>The <strong>sample_id</strong> is provided as input by the user, and reflects a
unique identifier of the tumor-normal sample pair to be analyzed.</p>
<ul class="simple">
<li><a class="reference external" href="http://folk.uio.no/sigven/tumor_sample.BRCA.pcgr.html">View an example report for a breast tumor sample
(TCGA)</a></li>
<li><a class="reference external" href="http://folk.uio.no/sigven/tumor_sample.COAD.pcgr.html">View an example report for a colorectal tumor sample
(TCGA)</a></li>
</ul>
<p>The HTML reports have been tested using the following browsers:</p>
<ul class="simple">
<li>Safari (10.0.3)</li>
<li>Mozilla Firefox (52.0.2)</li>
<li>Google Chrome (57.0.2987.110)</li>
</ul>
</div>
<div class="section" id="output-somatic-snvs-indels">
<h2>Output - Somatic SNVs/InDels<a class="headerlink" href="#output-somatic-snvs-indels" title="Permalink to this headline">¶</a></h2>
<div class="section" id="variant-call-format-vcf">
<h3>Variant call format - VCF<a class="headerlink" href="#variant-call-format-vcf" title="Permalink to this headline">¶</a></h3>
<p>A VCF file containing annotated, somatic calls (single nucleotide
variants and insertion/deletions) is generated with the following naming
convention:</p>
<p><strong>sample_id</strong>.pcgr.vcf.gz</p>
<p>Here, the <strong>sample_id</strong> is provided as input by the user, and reflects
a unique identifier of the tumor-normal sample pair to be analyzed.
Following common standards, the annotated VCF file is compressed with
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and indexed with
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a>. Below follows a
description of all annotations/tags present in the VCF INFO column after
processing with the PCGR annotation pipeline:</p>
<div class="section" id="vep-consequence-annotations">
<h4><em>VEP consequence annotations</em><a class="headerlink" href="#vep-consequence-annotations" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>CSQ - Complete consequence annotations from VEP. Format:
Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|
EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|
Codons|Existing_variation|ALLELE_NUM|DISTANCE|STRAND|FLAGS|PICK|VARIANT_CLASS|
SYMBOL_SOURCE|HGNC_ID|CANONICAL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|
UNIPARC|RefSeq|DOMAINS|HGVS_OFFSET|CLIN_SIG|SOMATIC|PHENO|MOTIF_NAME|
MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE</li>
<li>Consequence - Impact modifier for the consequence type (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Gene - Ensembl stable ID of affected gene (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>Feature_type - Type of feature. Currently one of Transcript,
RegulatoryFeature, MotifFeature (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>Feature - Ensembl stable ID of feature (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>cDNA_position - Relative position of base pair in cDNA sequence
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CDS_position - Relative position of base pair in coding sequence
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CDS_CHANGE - Coding, transcript-specific sequence annotation (picked
by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Protein_position - Relative position of amino acid in protein
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Amino_acids - Only given if the variant affects the protein-coding
sequence (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>Codons - The alternative codons with the variant base in upper case
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>IMPACT - Impact modifier for the consequence type (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>VARIANT_CLASS - Sequence Ontology variant class (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>SYMBOL - Gene symbol (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>SYMBOL_SOURCE - The source of the gene symbol (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>STRAND - The DNA strand (1 or -1) on which the transcript/feature
lies (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>ENSP - The Ensembl protein identifier of the affected transcript
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>FLAGS - Transcript quality flags: cds_start_NF: CDS 5&#8217;, incomplete
cds_end_NF: CDS 3&#8217; incomplete (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>SWISSPROT - Best match UniProtKB/Swiss-Prot accession of protein
product (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>TREMBL - Best match UniProtKB/TrEMBL accession of protein product
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>UNIPARC - Best match UniParc accession of protein product (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HGVSc - The HGVS coding sequence name (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>HGVSp - The HGVS protein sequence name (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>HGVSp_short - The HGVS protein sequence name, short version (picked
by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HGVS_OFFSET - Indicates by how many bases the HGVS notations for
this variant have been shifted (picked by VEP&#8217;s &#8211;flag_pick_allele
option)</li>
<li>MOTIF_NAME - The source and identifier of a transcription factor
binding profile aligned at this position (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>MOTIF_POS - The relative position of the variation in the aligned
TFBP (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>HIGH_INF_POS - A flag indicating if the variant falls in a high
information position of a transcription factor binding profile (TFBP)
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>MOTIF_SCORE_CHANGE - The difference in motif score of the reference
and variant sequences for the TFBP (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>CELL_TYPE - List of cell types and classifications for regulatory
feature (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>CANONICAL - A flag indicating if the transcript is denoted as the
canonical transcript for this gene (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>CCDS - The CCDS identifier for this transcript, where applicable
(picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>INTRON - The intron number (out of total number) (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>EXON - The exon number (out of total number) (picked by VEP&#8217;s
&#8211;flag_pick_allele option)</li>
<li>DOMAINS - The source and identifier of any overlapping protein
domains (picked by VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>DISTANCE - Shortest distance from variant to transcript (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>BIOTYPE - Biotype of transcript or regulatory feature (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>TSL - Transcript support level (picked by VEP&#8217;s &#8211;flag_pick_allele
option)&gt;</li>
<li>PUBMED - PubMed ID(s) of publications that cite existing variant -
VEP</li>
<li>PHENO - Indicates if existing variant is associated with a phenotype,
disease or trait - VEP</li>
<li>GENE_PHENO - Indicates if overlapped gene is associated with a
phenotype, disease or trait - VEP</li>
<li>ALLELE_NUM - Allele number from input; 0 is reference, 1 is first
alternate etc - VEP</li>
<li>REFSEQ_MATCH - The RefSeq transcript match status; contains a number
of flags indicating whether this RefSeq transcript matches the
underlying reference sequence and/or an Ensembl transcript (picked by
VEP&#8217;s &#8211;flag_pick_allele option)</li>
<li>PICK - Indicates if this block of consequence data was picked by
VEP&#8217;s &#8211;flag_pick_allele option</li>
<li>VEP_ALL_CONSEQUENCE - All transcript consequences
(Consequence:SYMBOL:Feature_type:Feature:BIOTYPE) - VEP</li>
</ul>
</div>
<div class="section" id="gene-information">
<h4><em>Gene information</em><a class="headerlink" href="#gene-information" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>ENTREZ_ID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/gene">Entrez</a> gene
identifier</li>
<li>APPRIS - Principal isoform flags according to the <a class="reference external" href="http://appris.bioinfo.cnio.es/#/downloads">APPRIS principal
isoform database</a></li>
<li>UNIPROT_ID - <a class="reference external" href="http://www.uniprot.org">UniProt</a> identifier</li>
<li>CANCER_CENSUS_SOMATIC - Gene with known cancer association -
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/census/">Cancer Gene Census,
WTSI</a></li>
<li>CANCER_CENSUS_GERMLINE - Gene with known cancer association -
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/census/">Cancer Gene Census,
WTSI</a></li>
<li>TUMOR_SUPPRESSOR - Gene is predicted as tumor suppressor candidate
according to (<a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene
v2.0</a>)</li>
<li>ONCOGENE - Gene is curated as an oncogene according to (<a class="reference external" href="http://bioinfo.mc.vanderbilt.edu/TSGene/">TSGene
v2.0</a>)</li>
<li>ONCOSCORE - Literature-derived score for cancer gene relevance
<a class="reference external" href="http://bioconductor.org/packages/release/bioc/html/OncoScore.html">Bioconductor/OncoScore</a>,
range from 0 (low oncogenic potential) to 1 (high oncogenic
potential)</li>
<li>INTOGEN_DRIVER - Gene is predicted as a cancer driver in the
<a class="reference external" href="https://www.intogen.org/downloads">IntoGen Cancer Drivers Database -
2014.12</a></li>
</ul>
</div>
<div class="section" id="variant-effect-and-protein-coding-information">
<h4><em>Variant effect and protein-coding information</em><a class="headerlink" href="#variant-effect-and-protein-coding-information" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>CANCER_MUTATION_HOTSPOT - mutation hotspot codon in
<a class="reference external" href="http://cancerhotspots.org/">cancerhotspots.org</a>. Format:
gene_symbol | codon | q-value</li>
<li>UNIPROT_FEATURE - Overlapping protein annotations from <a class="reference external" href="http://www.uniprot.org">UniProt
KB</a></li>
<li>INTOGEN_DRIVER_MUT - Indicates if existing variant is predicted as
driver mutation from IntoGen Catalog of Driver Mutations</li>
<li>EFFECT_PREDICTIONS - Predictions of effect of variant on protein
function and pre-mRNA splicing from <a class="reference external" href="https://sites.google.com/site/jpopgen/dbNSFP">database of non-synonymous
functional predictions - dbNSFP
v3.4</a>. Predicted
effects are provided by different sources/algorithms (separated by
&#8216;&amp;&#8217;):<ol class="arabic">
<li><a class="reference external" href="http://provean.jcvi.org/index.php">SIFT</a> (Jan 2015)</li>
<li><a class="reference external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen2-HDIV</a> (v
2.2.2)</li>
<li><a class="reference external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen2-HVAR</a> (v
2.2.2)</li>
<li><a class="reference external" href="http://www.genetics.wustl.edu/jflab/lrt_query.html">LRT</a>
(2009)</li>
<li><a class="reference external" href="http://www.mutationtaster.org/">MutationTaster</a> (data release
Nov 2015)</li>
<li><a class="reference external" href="http://mutationassessor.org/">MutationAssessor</a> (release 3)</li>
<li>[FATHMM] (<a class="reference external" href="http://fathmm.biocompute.org.uk">http://fathmm.biocompute.org.uk</a>) (v2.3)</li>
<li><a class="reference external" href="http://provean.jcvi.org/index.php">PROVEAN</a> (v1.1 Jan 2015)</li>
<li><a class="reference external" href="http://fathmm.biocompute.org.uk/fathmmMKL.htm">FATHMM_MKL</a></li>
<li><a class="reference external" href="http://cadd.gs.washington.edu/">CADD</a> (v1.3)</li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/25552646">DBNSFP_CONSENSUS_SVM</a>
(Ensembl/consensus prediction, based on support vector machines)</li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/25552646">DBNSFP_CONSENSUS_LR</a>
(Ensembl/consensus prediction, logistic regression based)</li>
<li><a class="reference external" href="http://nar.oxfordjournals.org/content/42/22/13534">SPLICE_SITE_EFFECT_ADA</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on
adaptive boosting)</li>
<li><a class="reference external" href="http://nar.oxfordjournals.org/content/42/22/13534">SPLICE_SITE_EFFECT_RF</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on
adaptive boosting)</li>
<li><a class="reference external" href="http://bejerano.stanford.edu/MCAP">M-CAP</a></li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27666373">REVEL</a></li>
<li><a class="reference external" href="http://mutpred.mutdb.org">MutPred</a></li>
<li><a class="reference external" href="http://mendel.stanford.edu/SidowLab/downloads/gerp/">GERP</a></li>
</ol>
</li>
</ul>
</div>
<div class="section" id="variant-frequencies-annotations-in-germline-somatic-databases">
<h4><em>Variant frequencies/annotations in germline/somatic databases</em><a class="headerlink" href="#variant-frequencies-annotations-in-germline-somatic-databases" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>AFR_AF_EXAC - African/American germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
1</a>)</li>
<li>AMR_AF_EXAC - American germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
1</a>)</li>
<li>GLOBAL_AF_EXAC - Adjusted global germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
1</a>)</li>
<li>EAS_AF_EXAC - East Asian germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
1</a>)</li>
<li>FIN_AF_EXAC - Finnish germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation
Consortium release 1</a>)</li>
<li>NFE_AF_EXAC - Non-Finnish European germline allele frequency
(<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation Consortium release
1</a>)</li>
<li>OTH_AF_EXAC - Other germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome Aggregation
Consortium release 1</a>)</li>
<li>SAS_AF_EXAC - South Asian germline allele frequency (<a class="reference external" href="http://exac.broadinstitute.org/">Exome
Aggregation Consortium release
1</a>)</li>
<li>AFR_AF_GNOMAD - African/American germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
1</a>)</li>
<li>AMR_AF_GNOMAD - American germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
1</a>)</li>
<li>GLOBAL_AF_GNOMAD - Adjusted global germline allele frequency
(<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome Aggregation Database release
1</a>)</li>
<li>SAS_AF_GNOMAD - South Asian germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
1</a>)</li>
<li>EAS_AF_GNOMAD - East Asian germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
11</a>)</li>
<li>FIN_AF_GNOMAD - Finnish germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
1</a>)</li>
<li>NFE_AF_GNOMAD - Non-Finnish European germline allele frequency
(<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome Aggregation Database release
1</a>)</li>
<li>OTH_AF_GNOMAD - Other germline allele frequency (<a class="reference external" href="http://gnomad.broadinstitute.org/">Genome
Aggregation Database release
1</a>)</li>
<li>AFR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from AFR (African)</li>
<li>AMR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from AMR (Ad Mixed American)</li>
<li>EAS_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from EAS (East Asian)</li>
<li>EUR_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from EUR (European)</li>
<li>SAS_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for
samples from SAS (South Asian)</li>
<li>GLOBAL_AF_1KG - <a class="reference external" href="http://www.1000genomes.org">1000G Project - phase
3</a> germline allele frequency for all
1000G project samples (global)</li>
<li>DBSNPRSID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a> reference ID</li>
<li>DBSNPBUILDID - Initial <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a>
build ID for rsID</li>
<li>DBSNP_MAPPINGSTATUS - Status with respect to the genomic mappability
of the flanking sequence of the rsID</li>
<li>DBSNP_VALIDATION - Categories of evidence that support the variant
in <a class="reference external" href="http://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a></li>
<li>DBSNP_SUBMISSIONS - Number of individual submissions to rsID</li>
<li>GWAS_CATALOG_PMID - Variant is linked to phenotype through the
<a class="reference external" href="https://www.ebi.ac.uk/gwas/">GWAS Catalog</a>, literature in PMID
list</li>
<li>GWAS_CATALOG_TRAIT_URI - List of trait URIs for GWAS-associated
variant</li>
<li>COSMIC_MUTATION_ID - Mutation identifier in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in
cancer</a>
database</li>
<li>COSMIC_CODON_FRAC_GW - For different tumor types, number of
samples mutated at associated codon position (format:
codon_number:tumor_type:fraction_mutated). Samples subject to
exome/genome-wide screens only <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a>.</li>
<li>COSMIC_CODON_COUNT_GW - For different tumor types, number of
samples mutated at associated codon position (format:
codon_number:tumor_type:frequency). Samples subject to
exome/genome-wide screens only <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a></li>
<li>COSMIC_COUNT_GW - Global frequency of variant in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of
somatic mutations in
cancer</a>.</li>
<li>COSMIC_SITE_HISTOLOGY - Primary site/histology distribution across
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a>.</li>
<li>COSMIC_CANCER_TYPE_GW - Frequency of variant across different
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a>
- samples subject to exome/genome-wide screens only</li>
<li>COSMIC_CANCER_TYPE_ALL - Frequency of variant across different
tumor types in <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a></li>
<li>COSMIC_SAMPLE_SOURCE - Sample source distribution for variant in
<a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a>.</li>
<li>COSMIC_DRUG_RESISTANCE - Targeted drugs/therapies subject to
resistance in tumors that carry the mutation. <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in
cancer</a>.</li>
<li>COSMIC_FATHMM_PRED - Variant effect prediction from COSMIC&#8217;s FATHMM
algorithm (COSMIC variants only) <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations in
cancer</a>.</li>
<li>COSMIC_VARTYPE - COSMIC variant type <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic mutations
in
cancer</a>.</li>
<li>COSMIC_CONSEQUENCE - COSMIC consequence type <a class="reference external" href="http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/">Catalog of somatic
mutations in
cancer</a>.</li>
<li>ICGC_PROJECTS - Variant frequency count in different <a class="reference external" href="https://dcc.icgc.org/repository/current/Projects">ICGC Project
IDs</a></li>
</ul>
</div>
<div class="section" id="clinical-associations">
<h4><em>Clinical associations</em><a class="headerlink" href="#clinical-associations" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>CLINVAR_MSID - <a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a>
Measure Set/Variant ID</li>
<li>CLINVAR_PMIDS - Associated Pubmed IDs for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>CLINVAR_SIG - Clinical significance for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>CLINVAR_VARIANT_ORIGIN - Origin of variant (somatic, germline, de
novo etc.) for variant in
<a class="reference external" href="http://www.ncbi.nlm.nih.gov/clinvar">ClinVar</a></li>
<li>DOCM_DISEASE - Associated disease types for variant in <a class="reference external" href="http://docm.genome.wustl.edu">Database of
Curated Mutations</a></li>
<li>DOCM_PMID - Associated Pubmed IDs for variant in <a class="reference external" href="http://docm.genome.wustl.edu">Database of
Curated Mutations</a></li>
</ul>
</div>
<div class="section" id="other">
<h4><em>Other</em><a class="headerlink" href="#other" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>ANTINEOPLASTIC_DRUG_INTERACTION - Approved and experimental
antineoplastic drugs interacting with the mutated gene, as retrieved
from the <a class="reference external" href="http://dgidb.genome.wustl.edu/">Drug-Gene Interaction
Database</a></li>
<li>CIVIC_ID, CIVIC_ID_2 - Variant identifiers in the <a class="reference external" href="http://civic.genome.wustl.edu">CIViC
database</a></li>
<li>CBMDB_ID - Variant identifier in the <a class="reference external" href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer bioMarkers
database</a></li>
</ul>
</div>
</div>
<div class="section" id="tab-separated-values-tsv">
<h3>Tab-separated values (TSV)<a class="headerlink" href="#tab-separated-values-tsv" title="Permalink to this headline">¶</a></h3>
<div class="section" id="annotated-list-of-all-snvs-indels">
<h4>Annotated List of all SNVs/InDels<a class="headerlink" href="#annotated-list-of-all-snvs-indels" title="Permalink to this headline">¶</a></h4>
<p>We provide a tab-separated values file with most important annotations
for SNVs/InDels. The file has the following naming convention:</p>
<p><strong>sample_id</strong>.pcgr.snvs_indels.tiers.tsv</p>
<p>The SNVs/InDels are organized into different <strong>tiers</strong> that reflect
relevance for therapeutics/tumorigenesis:</p>
<ul class="simple">
<li><strong>Tier 1</strong> constitute variants recorded as prognostic/diagnostic/drug
sensitivity biomarkers in the <a class="reference external" href="http://civic.genome.wustl.edu">CIViC
database</a> and the <a class="reference external" href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer
Biomarkers
Database</a></li>
<li><strong>Tier 2</strong> includes other coding variants that are found in known
mutational hotspots, predicted as cancer driver mutations, or curated
as disease-causing</li>
<li><strong>Tier 3</strong> includes other coding variants found in oncogenes, tumor
suppressor genes, or cancer census genes</li>
<li><strong>Tier 4</strong> includes other coding variants</li>
<li><strong>Tier 5</strong> includes non-coding variants</li>
</ul>
<p><strong>Note</strong>: &#8216;<em>coding variants</em>&#8216; refer to the set of variants with the
following consequences: - missense variant - splice donor/splice
acceptor alteration - stop gained/stop lost - frameshift/non-frameshift
variants</p>
<p>The following variables are included in the tiered TSV file:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">GENOMIC_CHANGE</span> <span class="o">-</span> <span class="n">Identifier</span> <span class="k">for</span> <span class="n">genomic</span> <span class="n">variant</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">g</span><span class="o">.</span><span class="n">chr1</span><span class="p">:</span><span class="mi">152382569</span><span class="p">:</span><span class="n">A</span><span class="o">&gt;</span><span class="n">G</span>
<span class="mf">2.</span> <span class="n">GENOME_VERSION</span> <span class="o">-</span> <span class="n">Assembly</span> <span class="n">version</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">GRCh37</span>
<span class="mf">3.</span> <span class="n">VCF_SAMPLE_ID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
<span class="mf">4.</span> <span class="n">VARIANT_CLASS</span> <span class="o">-</span> <span class="n">Variant</span> <span class="nb">type</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">SNV</span><span class="o">/</span><span class="n">insertion</span><span class="o">/</span><span class="n">deletion</span>
<span class="mf">5.</span> <span class="n">SYMBOL</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">symbol</span>
<span class="mf">6.</span> <span class="n">GENE_NAME</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">description</span>
<span class="mf">7.</span> <span class="n">CCDS</span> <span class="o">-</span> <span class="n">CCDS</span> <span class="n">identifier</span>
<span class="mf">8.</span> <span class="n">ENTREZ_ID</span> <span class="o">-</span> <span class="n">Entrez</span> <span class="n">gene</span> <span class="n">identifier</span>
<span class="mf">9.</span> <span class="n">UNIPROT_ID</span> <span class="o">-</span> <span class="n">UniProt</span> <span class="n">protein</span> <span class="n">identifier</span>
<span class="mf">10.</span> <span class="n">ONCOSCORE</span> <span class="o">-</span> <span class="n">Literature</span><span class="o">-</span><span class="n">derived</span> <span class="n">score</span> <span class="k">for</span> <span class="n">cancer</span> <span class="n">gene</span> <span class="n">relevance</span>
<span class="mf">11.</span> <span class="n">ONCOGENE</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">curated</span> <span class="k">as</span> <span class="n">an</span> <span class="n">oncogene</span> <span class="n">according</span> <span class="n">to</span> <span class="n">TSGene</span>
<span class="mf">12.</span> <span class="n">TUMOR_SUPPRESSOR</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span> <span class="n">tumor</span> <span class="n">suppressor</span>
    <span class="n">candidate</span> <span class="n">according</span> <span class="n">to</span> <span class="n">TSGene</span>
<span class="mf">13.</span> <span class="n">INTOGEN_DRIVER</span> <span class="o">-</span> <span class="n">Gene</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span> <span class="n">a</span> <span class="n">cancer</span> <span class="n">driver</span> <span class="ow">in</span> <span class="n">the</span>
    <span class="n">IntoGen</span> <span class="n">Cancer</span> <span class="n">Drivers</span> <span class="n">Database</span> <span class="o">-</span> <span class="mf">2014.12</span>
<span class="mf">14.</span> <span class="n">CANCER_CENSUS_SOMATIC</span> <span class="o">-</span> <span class="n">Gene</span> <span class="k">with</span> <span class="n">known</span> <span class="n">cancer</span> <span class="n">association</span> <span class="o">-</span>
    <span class="n">Cancer</span> <span class="n">Gene</span> <span class="n">Census</span><span class="p">,</span> <span class="n">WTSI</span>
<span class="mf">15.</span> <span class="n">CANCER_CENSUS_GERMLINE</span> <span class="o">-</span> <span class="n">Gene</span> <span class="k">with</span> <span class="n">known</span> <span class="n">cancer</span> <span class="n">association</span> <span class="o">-</span>
    <span class="n">Cancer</span> <span class="n">Gene</span> <span class="n">Census</span><span class="p">,</span> <span class="n">WTSI</span>
<span class="mf">16.</span> <span class="n">CONSEQUENCE</span> <span class="o">-</span> <span class="n">Variant</span> <span class="n">consequence</span> <span class="p">(</span><span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span> <span class="n">output</span><span class="p">:</span>
    <span class="n">Consequence</span><span class="p">)</span>
<span class="mf">17.</span> <span class="n">PROTEIN_CHANGE</span> <span class="o">-</span> <span class="n">Protein</span> <span class="n">change</span> <span class="p">(</span><span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span> <span class="n">output</span><span class="p">:</span>
    <span class="n">HGVSp_short</span><span class="p">)</span>
<span class="mf">18.</span> <span class="n">PROTEIN_DOMAIN</span> <span class="o">-</span> <span class="n">Protein</span> <span class="n">domain</span>
<span class="mf">19.</span> <span class="n">CDS_CHANGE</span> <span class="o">-</span> <span class="n">composite</span> <span class="n">variable</span> <span class="k">for</span> <span class="n">coding</span> <span class="n">change</span><span class="p">,</span> <span class="nb">format</span><span class="p">:</span>
    <span class="n">Consequence</span><span class="p">:</span><span class="n">Feature</span><span class="p">:</span><span class="n">cDNA_position</span><span class="p">:</span><span class="n">EXON</span><span class="p">:</span><span class="n">HGVSp_short</span>
<span class="mf">20.</span> <span class="n">EFFECT_PREDICTIONS</span> <span class="o">-</span> <span class="k">as</span> <span class="n">defined</span> <span class="n">above</span> <span class="k">for</span> <span class="n">VCF</span>
<span class="mf">21.</span> <span class="n">CANCER_MUTATION_HOTSPOT</span> <span class="o">-</span> <span class="n">mutation</span> <span class="n">hotspot</span> <span class="n">codon</span> <span class="ow">in</span>
    <span class="n">cancerhotspots</span><span class="o">.</span><span class="n">org</span><span class="o">.</span> <span class="n">Format</span><span class="p">:</span> <span class="n">gene_symbol</span> <span class="o">|</span> <span class="n">codon</span> <span class="o">|</span> <span class="n">q</span><span class="o">-</span><span class="n">value</span>
<span class="mf">22.</span> <span class="n">INTOGEN_DRIVER_MUT</span> <span class="o">-</span> <span class="n">Indicates</span> <span class="k">if</span> <span class="n">existing</span> <span class="n">variant</span> <span class="ow">is</span> <span class="n">predicted</span> <span class="k">as</span>
    <span class="n">driver</span> <span class="n">mutation</span> <span class="kn">from</span> <span class="nn">IntoGen</span> <span class="n">Catalog</span> <span class="n">of</span> <span class="n">Driver</span> <span class="n">Mutations</span>
<span class="mf">23.</span> <span class="n">VEP_ALL_CONSEQUENCE</span> <span class="o">-</span> <span class="nb">all</span> <span class="n">VEP</span> <span class="n">consequences</span>
<span class="mf">24.</span> <span class="n">DBSNP</span> <span class="o">-</span> <span class="n">dbSNP</span> <span class="n">reference</span> <span class="n">cluster</span> <span class="n">ID</span>
<span class="mf">25.</span> <span class="n">COSMIC</span> <span class="o">-</span> <span class="n">COSMIC</span> <span class="n">mutation</span> <span class="n">ID</span>
<span class="mf">26.</span> <span class="n">COSMIC_SITE_HISTOLOGY</span> <span class="o">-</span> <span class="n">distribution</span> <span class="n">of</span> <span class="n">tumor</span> <span class="n">sites</span><span class="o">/</span><span class="n">histology</span> <span class="n">types</span>
    <span class="k">for</span> <span class="n">COSMIC</span> <span class="n">mutation</span>
<span class="mf">27.</span> <span class="n">COSMIC_DRUG_RESISTANCE</span> <span class="o">-</span> <span class="n">variant</span> <span class="n">associated</span> <span class="k">with</span> <span class="n">resistance</span> <span class="n">to</span> <span class="n">a</span>
    <span class="n">particular</span> <span class="n">antineoplastic</span> <span class="n">drug</span>
<span class="mf">28.</span> <span class="n">CLINVAR</span> <span class="o">-</span> <span class="n">variant</span> <span class="n">origin</span> <span class="ow">and</span> <span class="n">associated</span> <span class="n">traits</span> <span class="n">associated</span> <span class="k">with</span> <span class="n">variant</span>
<span class="mf">29.</span> <span class="n">CLINVAR_SIG</span> <span class="o">-</span> <span class="n">clinical</span> <span class="n">significance</span> <span class="n">of</span> <span class="n">CLINVAR</span> <span class="n">variant</span>
<span class="mf">30.</span> <span class="n">GLOBAL_AF_EXAC</span> <span class="o">-</span> <span class="n">adjusted</span> <span class="k">global</span> <span class="n">germline</span> <span class="n">allele</span> <span class="n">frequency</span> <span class="ow">in</span> <span class="n">ExAC</span>
<span class="mf">31.</span> <span class="n">GLOBAL_AF_1KG</span> <span class="o">-</span> <span class="mi">1000</span><span class="n">G</span> <span class="n">Project</span> <span class="o">-</span> <span class="n">phase</span> <span class="mi">3</span><span class="p">,</span> <span class="n">germline</span> <span class="n">allele</span> <span class="n">frequency</span>
    <span class="k">for</span> <span class="nb">all</span> <span class="mi">1000</span><span class="n">G</span> <span class="n">project</span> <span class="n">samples</span> <span class="p">(</span><span class="k">global</span><span class="p">)</span>
<span class="mf">32.</span> <span class="n">TIER</span>
<span class="mf">33.</span> <span class="n">TIER_DESCRIPTION</span>
</pre></div>
</div>
</div>
<div class="section" id="biomarkers-among-snvs-indels">
<h4>Biomarkers among SNVs/InDEls<a class="headerlink" href="#biomarkers-among-snvs-indels" title="Permalink to this headline">¶</a></h4>
<p>For tumor samples that have variant hits in <strong>Tier 1</strong> we provide an
additional file with all associated <a class="reference external" href="https://civic.genome.wustl.edu/#/help/evidence/overview">clinical evidence
items</a>. The
file has the following naming convention:</p>
<p><strong>sample_id</strong>.pcgr.snvs_indels.biomarkers.tsv</p>
<p>The format of the biomarker TSV file is as follows:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">GENOMIC_CHANGE</span> <span class="o">-</span> <span class="n">Identifier</span> <span class="k">for</span> <span class="n">genomic</span> <span class="n">variant</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">g</span><span class="o">.</span><span class="n">chr1</span><span class="p">:</span><span class="mi">152382569</span><span class="p">:</span><span class="n">A</span><span class="o">&gt;</span><span class="n">G</span>
<span class="mf">2.</span> <span class="n">GENOME_VERSION</span> <span class="o">-</span> <span class="n">Assembly</span> <span class="n">version</span><span class="p">,</span> <span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span> <span class="n">GRCh37</span>
<span class="mf">3.</span> <span class="n">VCF_SAMPLE_ID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
<span class="mf">4.</span> <span class="n">SYMBOL</span> <span class="o">-</span> <span class="n">Gene</span> <span class="n">symbol</span>
<span class="mf">5.</span> <span class="n">CONSEQUENCE</span> <span class="o">-</span> <span class="n">Variant</span> <span class="n">consequence</span>
<span class="mf">6.</span> <span class="n">BM_CLINICAL_SIGNIFICANCE</span> <span class="o">-</span> <span class="n">The</span> <span class="n">association</span> <span class="k">with</span> <span class="n">diagnostic</span><span class="o">/</span><span class="n">prognostic</span> <span class="n">end</span> <span class="n">point</span> <span class="ow">or</span> <span class="n">treatment</span>
<span class="mf">7.</span> <span class="n">BM_EVIDENCE_LEVEL</span> <span class="o">-</span> <span class="n">The</span> <span class="nb">type</span> <span class="n">of</span> <span class="n">experiment</span> <span class="kn">from</span> <span class="nn">which</span> <span class="n">the</span> <span class="n">evidence</span> <span class="ow">is</span> <span class="n">curated</span> <span class="p">(</span><span class="n">validated</span><span class="p">,</span> <span class="n">clinical</span><span class="p">,</span> <span class="n">pre</span><span class="o">-</span><span class="n">clinical</span><span class="p">,</span> <span class="n">case</span> <span class="n">study</span><span class="p">,</span> <span class="ow">and</span> <span class="n">inferential</span><span class="p">)</span>
<span class="mf">8.</span> <span class="n">BM_EVIDENCE_TYPE</span> <span class="o">-</span> <span class="n">Category</span> <span class="n">of</span> <span class="n">clinical</span> <span class="n">action</span><span class="o">/</span><span class="n">relevance</span> <span class="n">implicated</span> <span class="n">by</span> <span class="n">event</span> <span class="p">(</span><span class="n">Predictive</span><span class="p">,</span> <span class="n">Prognostic</span><span class="p">,</span> <span class="n">Predisposing</span> <span class="ow">and</span> <span class="n">Diagnostic</span><span class="p">)</span>
<span class="mf">9.</span> <span class="n">BM_EVIDENCE_DIRECTION</span> <span class="o">-</span> <span class="n">An</span> <span class="n">indicator</span> <span class="n">of</span> <span class="n">whether</span> <span class="n">the</span> <span class="n">evidence</span> <span class="n">statement</span> <span class="n">supports</span> <span class="ow">or</span> <span class="n">refutes</span> <span class="n">the</span> <span class="n">clinical</span> <span class="n">significance</span> <span class="n">of</span> <span class="n">an</span> <span class="n">event</span>
<span class="mf">10.</span> <span class="n">BM_DISEASE_NAME</span> <span class="o">-</span> <span class="n">Specific</span> <span class="n">disease</span> <span class="ow">or</span> <span class="n">disease</span> <span class="n">subtype</span> <span class="n">that</span> <span class="ow">is</span> <span class="n">associated</span> <span class="k">with</span> <span class="n">this</span> <span class="n">event</span> <span class="ow">and</span> <span class="n">its</span> <span class="n">clinical</span> <span class="n">implication</span>
<span class="mf">11.</span> <span class="n">BM_DRUG_NAMES</span> <span class="o">-</span> <span class="n">For</span> <span class="n">predictive</span> <span class="n">evidence</span><span class="p">,</span> <span class="n">indicates</span> <span class="n">the</span> <span class="n">therapy</span> <span class="k">for</span> <span class="n">which</span> <span class="n">sensitivity</span> <span class="ow">or</span> <span class="n">resistance</span> <span class="ow">is</span> <span class="n">indicated</span>
<span class="mf">12.</span> <span class="n">BM_RATING</span> <span class="o">-</span> <span class="n">A</span> <span class="n">rating</span> <span class="n">on</span> <span class="n">a</span> <span class="mi">5</span><span class="o">-</span><span class="n">star</span> <span class="n">scale</span><span class="p">,</span> <span class="n">portraying</span> <span class="n">the</span> <span class="n">curators</span> <span class="n">trust</span> <span class="ow">in</span> <span class="n">the</span> <span class="n">experiments</span> <span class="kn">from</span> <span class="nn">which</span> <span class="n">the</span> <span class="n">evidence</span> <span class="ow">is</span> <span class="n">curated</span>
<span class="mf">13.</span> <span class="n">BM_CITATION</span> <span class="o">-</span> <span class="n">Publication</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="n">where</span> <span class="n">the</span> <span class="n">event</span> <span class="n">was</span> <span class="n">described</span><span class="o">/</span><span class="n">explored</span><span class="o">/</span><span class="n">guidelines</span><span class="o">/</span><span class="n">trials</span>
<span class="mf">14.</span> <span class="n">TIER</span>
<span class="mf">15.</span> <span class="n">TIER_DESCRIPTION</span>
</pre></div>
</div>
</div>
<div class="section" id="mutational-signatures">
<h4>Mutational signatures<a class="headerlink" href="#mutational-signatures" title="Permalink to this headline">¶</a></h4>
<p>For each tumor sample, we apply the <a class="reference external" href="https://github.com/raerose01/deconstructSigs">deconstructSigs
package</a> to delineate
the known mutational signatures. The inferred, weighted contributions by
each signature and their underlying, proposed etiologies are given in a
TSV file with the following naming convention:</p>
<p><strong>sample_id</strong>.pcgr.mutational_signatures.tsv</p>
<p>The format of the mutational signatures TSV file is as follows:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">Signature_ID</span> <span class="o">-</span> <span class="n">ID</span> <span class="n">of</span> <span class="n">signature</span> <span class="kn">from</span> <span class="nn">COSMIC</span><span class="s1">&#39;s 30 reference signatures</span>
<span class="mf">2.</span> <span class="n">Weight</span> <span class="o">-</span> <span class="n">inferred</span> <span class="n">weight</span> <span class="n">of</span> <span class="n">signature</span> <span class="ow">in</span> <span class="n">the</span> <span class="n">tumor</span> <span class="n">sample</span>
<span class="mf">3.</span> <span class="n">Cancer_types</span> <span class="o">-</span> <span class="n">cancer</span> <span class="n">types</span> <span class="ow">in</span> <span class="n">which</span> <span class="n">the</span> <span class="n">signature</span> <span class="n">has</span> <span class="n">been</span> <span class="n">observed</span>
<span class="mf">4.</span> <span class="n">Proposed_aetiology</span> <span class="o">-</span> <span class="n">proposed</span> <span class="n">underlying</span> <span class="n">etiology</span>
<span class="mf">5.</span> <span class="n">SampleID</span> <span class="o">-</span> <span class="n">Sample</span> <span class="n">identifier</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="output-somatic-copy-number-abberations">
<h2>Output - Somatic copy number abberations<a class="headerlink" href="#output-somatic-copy-number-abberations" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id1">
<h3>1. Tab-separated values (TSV)<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>Copy number segments are intersected with the genomic coordinates of all
transcripts from (<a class="reference external" href="https://www.gencodegenes.org/releases/25lift37.html">ENSEMBL/GENCODE&#8217;s basic gene
annotation</a>. In
adddition, we attach cancer-relevant annotations for the affected
transcripts. The naming convention of the compressed TSV file is as
follows:</p>
<p><strong>sample_id</strong>.pcgr.cna_segments.tsv.gz</p>
<p>The format of the compressed TSV file is the following:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">chrom</span> <span class="o">-</span> <span class="n">chromosome</span>
<span class="mf">2.</span> <span class="n">segment_start</span> <span class="o">-</span> <span class="n">start</span> <span class="n">of</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">segment</span>
<span class="mf">3.</span> <span class="n">segment_end</span> <span class="o">-</span> <span class="n">end</span> <span class="n">of</span> <span class="n">copy</span> <span class="n">number</span> <span class="n">segment</span>
<span class="mf">4.</span> <span class="n">segment_length</span> <span class="o">-</span> <span class="n">length</span> <span class="n">of</span> <span class="n">segment</span> <span class="ow">in</span> <span class="n">Mb</span>
<span class="mf">5.</span> <span class="n">LogR</span> <span class="o">-</span> <span class="n">Copy</span> <span class="n">log</span><span class="o">-</span><span class="n">ratio</span>
<span class="mf">6.</span> <span class="n">ensembl_gene_id</span>
<span class="mf">7.</span> <span class="n">symbol</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">symbol</span>
<span class="mf">8.</span> <span class="n">ensembl_transcript_id</span>
<span class="mf">9.</span> <span class="n">transcript_start</span>
<span class="mf">10.</span> <span class="n">transcript_end</span>
<span class="mf">11.</span> <span class="n">transcript_overlap_percent</span> <span class="o">-</span> <span class="n">percent</span> <span class="n">of</span> <span class="n">transcript</span> <span class="n">length</span> <span class="n">covered</span> <span class="n">by</span> <span class="n">CN</span> <span class="n">segment</span>
<span class="mf">12.</span> <span class="n">name</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">name</span> <span class="n">description</span>
<span class="mf">13.</span> <span class="n">gene_biotype</span> <span class="o">-</span> <span class="nb">type</span> <span class="n">of</span> <span class="n">gene</span>
<span class="mf">14.</span> <span class="n">cancer_census_germline</span> <span class="o">-</span> <span class="n">gene</span> <span class="n">implicated</span> <span class="k">with</span> <span class="n">germline</span> <span class="n">predisposition</span> <span class="n">to</span> <span class="n">various</span> <span class="n">cancer</span> <span class="n">subtypes</span>
<span class="mf">15.</span> <span class="n">cancer_census_somatic</span> <span class="o">-</span> <span class="n">gene</span> <span class="k">for</span> <span class="n">which</span> <span class="n">somatic</span> <span class="n">mutations</span> <span class="n">have</span> <span class="n">been</span> <span class="n">causally</span> <span class="n">implicated</span> <span class="ow">in</span> <span class="n">tumor</span> <span class="n">development</span>
<span class="mf">16.</span> <span class="n">tsgene</span> <span class="o">-</span> <span class="n">tumor</span> <span class="n">suppressor</span> <span class="n">gene</span> <span class="n">status</span> <span class="p">(</span><span class="n">TSgene</span> <span class="n">database</span><span class="p">)</span>
<span class="mf">17.</span> <span class="n">tsgene_oncogene</span> <span class="o">-</span> <span class="n">oncogene</span> <span class="n">status</span> <span class="p">(</span><span class="n">TSgene</span> <span class="n">database</span><span class="p">)</span>
<span class="mf">18.</span> <span class="n">intogen_drivers</span> <span class="o">-</span> <span class="n">predicted</span> <span class="n">driver</span> <span class="n">gene</span> <span class="n">status</span> <span class="p">(</span><span class="n">IntoGen</span> <span class="n">Cancer</span> <span class="n">Drivers</span> <span class="n">Database</span><span class="p">)</span>
<span class="mf">19.</span> <span class="n">antineoplastic_drugs_dgidb</span> <span class="o">-</span> <span class="n">validated</span> <span class="ow">and</span> <span class="n">experimental</span> <span class="n">antineoplastic</span> <span class="n">drugs</span> <span class="n">interacting</span> <span class="k">with</span> <span class="n">gene</span>
<span class="mf">20.</span> <span class="n">gencode_transcript_type</span> <span class="o">-</span>
<span class="mf">21.</span> <span class="n">gencode_tag</span> <span class="o">-</span>
<span class="mf">22.</span> <span class="n">gencode_v19</span> <span class="o">-</span> <span class="n">transcript</span> <span class="ow">is</span> <span class="n">part</span> <span class="n">of</span> <span class="n">GENCODE</span> <span class="n">V19</span>
</pre></div>
</div>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="annotation_resources.html" class="btn btn-neutral" title="Annotation resources" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'0.1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>