## Getting started

### Prerequisites

#### Installation of Docker

  * Running PCGR requires that Docker is set up on your host. Docker has very complete installation instructions for different platforms:

    - installing [Docker on Linux](https://docs.docker.com/engine/installation/linux/)
    - installing [Docker on Mac OS](https://docs.docker.com/engine/installation/mac/)

  * Check that Docker is running, e.g by typing `docker ps` or `docker images` in your terminal window

  * __IMPORTANT__: The following represent the _minimal_ computing resources that must be assigned to the Docker virtual machine:
     * Memory: 5GB
     * CPUs: 4

     For Docker version 1.13 on [Mac OS X](https://docs.docker.com/docker-for-mac/#advanced) there is an option to change CPU's and RAM from the UI and restart Docker. This can be found through [Docker Preferences (Advanced)](https://docs.docker.com/docker-for-windows/#advanced) in the toolbar. Similarly for [Docker for Windows](https://docs.docker.com/docker-for-windows/#advanced).

#### Python

An installation of Python (version 2.7.13) is required to run PCGR. Check that Python is installed by typing `python --version` in your terminal window.

#### Download PCGR

__August 4th 2017__: New release (0.4.0)</font>

* Download and unpack the [latest software release (0.4.0)](https://github.com/sigven/pcgr/releases/latest)

* Download and unpack the data bundle (approx. 17Gb) in the PCGR directory
    * Download [the latest data bundle](https://drive.google.com/file/d/0B8aYD2TJ472mNnpLOFNXdFV3bVE/) from Google Drive to `~/pcgr-X.X` (replace _X.X_ with the version number, e.g. `~/pcgr-0.4.0`)
    * Decompress and untar the bundle, e.g. through the following Unix command: `gzip -dc pcgr.databundle.GRCh37.YYYYMMDD.tgz | tar xvf -`
    * __IMPORTANT NOTE__: The data bundle contains elements (e.g. [COSMIC](http://cancer.sanger.ac.uk/cancergenome/assets/COSMIC_academic_license_march2015.pdf), [IntoGen driver mutations](https://www.intogen.org/downloads), [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP)) which are freely available for non-commercial use only

    A _data/_ folder within the _pcgr-X.X_ software folder should now have been produced

* Pull the [PCGR Docker image - 0.4.0](https://hub.docker.com/r/sigven/pcgr/) from DockerHub (3.3Gb) :
    * `docker pull sigven/pcgr:0.4.0` (PCGR annotation engine)


### Run test - generation of clinical report for a cancer genome

A tumor sample report is generated by calling the Python script __pcgr.py__, which takes the following arguments and options:

    usage: pcgr.py [-h] [--input_vcf INPUT_VCF] [--input_cna INPUT_CNA]
               [--logR_gain LOGR_GAIN] [--logR_homdel LOGR_HOMDEL]
               [--msig_identify] [--msig_n MSIG_N]
               [--msig_normalization {default,exome,genome,exome2genome}]
               [--msi_predict] [--list_noncoding]
               [--n_vcfanno_proc N_VCFANNO_PROC] [--n_vep_forks N_VEP_FORKS]
               [--tumor_dp_tag TUMOR_DP_TAG] [--tumor_af_tag TUMOR_AF_TAG]
               [--normal_dp_tag NORMAL_DP_TAG] [--normal_af_tag NORMAL_AF_TAG]
               [--call_conf_tag CALL_CONF_TAG] [--force_overwrite] [--version]
               pcgr_dir output_dir sample_id

    Personal Cancer Genome Reporter (PCGR) workflow for clinical interpretation of
    somatic nucleotide variants and copy number aberration segments

    positional arguments:
    pcgr_dir              PCGR base directory with accompanying data directory,
                        e.g. ~/pcgr-0.4.0
    output_dir            Output directory
    sample_id             Tumor sample/cancer genome identifier - prefix for
                        output files

    optional arguments:
    -h, --help            show this help message and exit
    --input_vcf INPUT_VCF
                        VCF input file with somatic query variants
                        (SNVs/InDels). Note: GRCh37 is currently the only
                        reference genome build supported (default: None)
    --input_cna INPUT_CNA
                        Somatic copy number alteration segments (tab-separated
                        values) (default: None)
    --logR_gain LOGR_GAIN
                        Log(2) ratio treshold for reporting copy number gains
                        in genome report (default: 0.8)
    --logR_homdel LOGR_HOMDEL
                        Log(2) ratio treshold for reporting homozygous
                        deletions in genome report (default: -0.8)
    --msig_identify       Identify relative contribution of 30 known mutational
                        signatures (COSMIC) through the deconstructSigs
                        framework (default: False)
    --msig_n MSIG_N       deconstructSigs option: number of mutational
                        signatures (to limit the search to (default: 6)
    --msig_normalization {default,exome,genome,exome2genome}
                        deconstructSigs option: type of trimer count
                        normalization for inference of known mutational
                        signatures, see explanation at
                        https://github.com/raerose01/deconstructSigs (default:
                        default)
    --msi_predict         Perform statistical prediction of tumor microsatellite
                        instability from distribution of somatic indels/snvs
                        in input VCF (default: False)
    --list_noncoding      List non-coding variants in HTML report (default:
                        False)
    --n_vcfanno_proc N_VCFANNO_PROC
                        Number of processes used during vcfanno annotation
                        (default: 4)
    --n_vep_forks N_VEP_FORKS
                        Number of forks (--forks option in VEP) used during
                        VEP annotation (default: 4)
    --tumor_dp_tag TUMOR_DP_TAG
                        Tag in input VCF (INFO column) that denotes total read
                        depth at variant site in tumor sample (default: _na)
    --tumor_af_tag TUMOR_AF_TAG
                        Tag in input VCF (INFO column) that denotes fraction
                        of alternate allele reads in tumor sample (default:
                        _na)
    --normal_dp_tag NORMAL_DP_TAG
                        Tag in input VCF (INFO column) that denotes total read
                        depth at variant site in control/normal sample
                        (default: _na)
    --normal_af_tag NORMAL_AF_TAG
                        Tag in input VCF (INFO column) that denotes fraction
                        of alternate allele reads control/normal sample
                        (default: _na)
    --call_conf_tag CALL_CONF_TAG
                        Tag in input VCF (INFO column) that denotes level of
                        confidence in variant call (default: _na)
    --force_overwrite     By default, the script will fail with an error if any
                        output file already exists. You can force the
                        overwrite of existing result files by using this flag
                        (default: False)
    --version             show program's version number and exit


The _examples_ folder contain input files from two tumor samples sequenced within TCGA. A report for a colorectal tumor case can be generated by running the following command in your terminal window:

`python pcgr.py --input_vcf ~/pcgr-0.4.0/examples/tumor_sample.COAD.vcf.gz --input_cna ~/pcgr-0.4.0/examples/tumor_sample.COAD.cna.tsv`
`--msig_identify --msi_predict --tumor_af_tag TVAF --tumor_dp_tag TDP --call_conf_tag TAL --list_noncoding`
`~/pcgr-0.4.0 ~/pcgr-0.4.0/examples tumor_sample.COAD`

This command will run the Docker-based PCGR workflow and produce the following output files in the _examples_ folder:

  1. __tumor_sample.COAD.pcgr.html__ - An interactive HTML report for clinical interpretation
  2. __tumor_sample.COAD.pcgr.vcf.gz__ - VCF file with rich set of annotations for precision oncology
  3.  __tumor_sample.COAD.pcgr.maf__ - A basic MAF file for use as input in downstream analyses with other tools (e.g. [2020plus](https://github.com/KarchinLab/2020plus), MutSigCV)
  4. __tumor_sample.COAD.pcgr.snvs_indels.tiers.tsv__ - Tab-separated values file with variants organized according to tiers of functional relevance
  5. __tumor_sample.COAD.pcgr.mutational_signatures.tsv__ - Tab-separated values file with estimated contributions by known mutational signatures and associated underlying etiologies
  6. __tumor_sample.COAD.pcgr.snvs_indels.biomarkers.tsv__ - Tab-separated values file with clinical evidence items associated with biomarkers for diagnosis, prognosis or drug sensitivity/resistance
  7. __tumor_sample.COAD.pcgr.cna_segments.tsv.gz__ - Tab-separated values file with annotations of gene transcripts that overlap with somatic copy number aberrations
